diff --git a/CMakeLists.txt b/CMakeLists.txt
index 01a40ee..02e775c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,6 +31,7 @@ link_directories(
 add_library(${PROJECT_NAME} SHARED
   src/gazebo_ros2_control_plugin.cpp
 )
+target_compile_definitions(${PROJECT_NAME} PRIVATE "_USE_MATH_DEFINES" "NO_STRICT")
 ament_target_dependencies(${PROJECT_NAME}
   angles
   controller_manager
@@ -43,6 +44,7 @@ ament_target_dependencies(${PROJECT_NAME}
   yaml_cpp_vendor
 )
 
+target_compile_definitions(gazebo_hardware_plugins PRIVATE "_USE_MATH_DEFINES" "NO_STRICT")
 add_library(gazebo_hardware_plugins SHARED
   src/gazebo_system.cpp
 )
