diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7e849fe..427c5cf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,6 +26,7 @@ find_package(ublox_ubx_msgs REQUIRED)
 find_package(ublox_ubx_interfaces REQUIRED)
 
 find_package(PkgConfig REQUIRED)
+pkg_check_modules(libudev REQUIRED libudev)
 pkg_check_modules(libusb REQUIRED libusb-1.0)
 
 include_directories(include SYSTEM)
@@ -45,11 +46,14 @@ ament_target_dependencies(ublox_dgnss_components
   std_msgs
   ublox_ubx_msgs
   ublox_ubx_interfaces
+  libusb
+  libudev
 )
 
-target_link_libraries(ublox_dgnss_components
-  usb-1.0
-)
+# target_link_libraries(ublox_dgnss_components
+#   ${libusb_LIBRARIES}
+#   ${libudev_LIBRARIES}
+# )
 
 rclcpp_components_register_node(ublox_dgnss_components PLUGIN "ublox_dgnss::UbloxDGNSSNode" EXECUTABLE ublox_dgnss_node)
 
