diff --git a/serial_driver/CMakeLists.txt b/serial_driver/CMakeLists.txt
index 4b15ca5..9055ace 100644
--- a/serial_driver/CMakeLists.txt
+++ b/serial_driver/CMakeLists.txt
@@ -24,6 +24,7 @@ endif()
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 ## dependencies
 find_package(ament_cmake_auto REQUIRED)
