diff --git a/src/occupancy_grid_node_main.cpp b/src/occupancy_grid_node_main.cpp
index 282b890c8..e5f527b2d 100644
--- a/src/occupancy_grid_node_main.cpp
+++ b/src/occupancy_grid_node_main.cpp
@@ -94,7 +94,8 @@ Node::Node(const double resolution, const double publish_period_sec)
   callback_group_executor_->add_callback_group(callback_group_, this->get_node_base_interface());
   client_ = this->create_client<cartographer_ros_msgs::srv::SubmapQuery>(
         kSubmapQueryServiceName,
-#if RCLCPP_VERSION_GTE(17, 0, 0)
+// Hardcoding this to false as we are in humble, hardcode to true on other distros
+#if false
         rclcpp::ServicesQoS(),
 #else
         rmw_qos_profile_services_default,
