diff --git a/CMakeLists.txt b/CMakeLists.txt
index a5284cad..8f925de2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -68,7 +68,7 @@ endfunction()
 
 ament_python_install_package(${PROJECT_NAME})
 
-pybind11_add_module(_reader SHARED
+add_library(_reader MODULE
   src/rosbag2_py/_reader.cpp
 )
 ament_target_dependencies(_reader PUBLIC
@@ -77,8 +77,15 @@ ament_target_dependencies(_reader PUBLIC
   "rosbag2_storage"
 )
 clean_windows_flags(_reader)
+set_target_properties(_reader PROPERTIES PREFIX "")
+if(WIN32)
+  set_target_properties(_reader PROPERTIES SUFFIX ".pyd")
+endif()
+target_link_libraries(_reader PRIVATE
+  pybind11::module
+)
 
-pybind11_add_module(_storage SHARED
+add_library(_storage MODULE
   src/rosbag2_py/_storage.cpp
   src/rosbag2_py/format_bag_metadata.cpp
 )
@@ -87,8 +91,15 @@ ament_target_dependencies(_storage PUBLIC
   "rosbag2_storage"
 )
 clean_windows_flags(_storage)
+set_target_properties(_storage PROPERTIES PREFIX "")
+if(WIN32)
+  set_target_properties(_storage PROPERTIES SUFFIX ".pyd")
+endif()
+target_link_libraries(_storage PRIVATE
+  pybind11::module
+)
 
-pybind11_add_module(_writer SHARED
+add_library(_writer MODULE
   src/rosbag2_py/_writer.cpp
 )
 ament_target_dependencies(_writer PUBLIC
@@ -97,8 +105,15 @@ ament_target_dependencies(_writer PUBLIC
   "rosbag2_storage"
 )
 clean_windows_flags(_writer)
+set_target_properties(_writer PROPERTIES PREFIX "")
+if(WIN32)
+  set_target_properties(_writer PROPERTIES SUFFIX ".pyd")
+endif()
+target_link_libraries(_writer PRIVATE
+  pybind11::module
+)
 
-pybind11_add_module(_info SHARED
+add_library(_info MODULE
   src/rosbag2_py/_info.cpp
 )
 ament_target_dependencies(_info PUBLIC
@@ -106,8 +118,15 @@ ament_target_dependencies(_info PUBLIC
   "rosbag2_storage"
 )
 clean_windows_flags(_info)
+set_target_properties(_info PROPERTIES PREFIX "")
+if(WIN32)
+  set_target_properties(_info PROPERTIES SUFFIX ".pyd")
+endif()
+target_link_libraries(_info PRIVATE
+  pybind11::module
+)
 
-pybind11_add_module(_transport SHARED
+add_library(_transport MODULE
   src/rosbag2_py/_transport.cpp
 )
 ament_target_dependencies(_transport PUBLIC
@@ -117,8 +133,15 @@ ament_target_dependencies(_transport PUBLIC
   "rosbag2_transport"
 )
 clean_windows_flags(_transport)
+set_target_properties(_transport PROPERTIES PREFIX "")
+if(WIN32)
+  set_target_properties(_transport PROPERTIES SUFFIX ".pyd")
+endif()
+target_link_libraries(_transport PRIVATE
+  pybind11::module
+)
 
-pybind11_add_module(_reindexer SHARED
+add_library(_reindexer MODULE
   src/rosbag2_py/_reindexer.cpp
 )
 ament_target_dependencies(_reindexer PUBLIC
@@ -126,6 +146,13 @@ ament_target_dependencies(_reindexer PUBLIC
   "rosbag2_storage"
 )
 clean_windows_flags(_reindexer)
+set_target_properties(_reindexer PROPERTIES PREFIX "")
+if(WIN32)
+  set_target_properties(_reindexer PROPERTIES SUFFIX ".pyd")
+endif()
+target_link_libraries(_reindexer PRIVATE
+  pybind11::module
+)
 
 # Install cython modules as sub-modules of the project
 install(
