From 61c6960edc787aa23c517f9a6be6e72d5a97f7be Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Fri, 17 Jun 2022 17:24:15 +0200
Subject: [PATCH] Update CMakeLists.txt

---
 moveit_core/CMakeLists.txt | 33 ++++++++++++++++++++++++---------
 1 file changed, 24 insertions(+), 9 deletions(-)

diff --git a/moveit_core/CMakeLists.txt b/moveit_core/CMakeLists.txt
index 9f000181c7..5e1f7de5f5 100644
--- a/moveit_core/CMakeLists.txt
+++ b/moveit_core/CMakeLists.txt
@@ -10,15 +10,30 @@ find_package(rclcpp REQUIRED)
 find_package(eigen3_cmake_module REQUIRED)
 find_package(Eigen3 REQUIRED)
 
-find_package(PkgConfig REQUIRED)
-pkg_check_modules(LIBFCL REQUIRED "fcl>=0.5.0")
-# replace LIBFCL_LIBRARIES with full paths to the libraries
-set(LIBFCL_LIBRARIES_FULL "")
-foreach(LIBFCL_LIBRARY ${LIBFCL_LIBRARIES})
-  find_library(${LIBFCL_LIBRARY}_LIB ${LIBFCL_LIBRARY} ${LIBFCL_LIBRARY_DIRS})
-  list(APPEND LIBFCL_LIBRARIES_FULL ${${LIBFCL_LIBRARY}_LIB})
-endforeach()
-set(LIBFCL_LIBRARIES "${LIBFCL_LIBRARIES_FULL}")
+find_package(fcl QUIET)
+if (fcl_FOUND)
+  # convert cmake target to variables for catkin_package DEPENDS
+  set(LIBFCL_FOUND TRUE)
+  if(TARGET ${FCL_LIBRARIES})
+    get_target_property(LIBFCL_LOCATION ${FCL_LIBRARIES} LOCATION)
+    get_target_property(LIBFCL_INTERFACE_LINK_LIBRARIES ${FCL_LIBRARIES} INTERFACE_LINK_LIBRARIES)
+    get_target_property(LIBFCL_INCLUDE_DIRS ${FCL_LIBRARIES} INTERFACE_INCLUDE_DIRECTORIES)
+    set(LIBFCL_LIBRARIES ${LIBFCL_LOCATION} ${LIBFCL_INTERFACE_LINK_LIBRARIES})
+  else()
+    set(LIBFCL_LIBRARIES ${FCL_LIBRARIES})
+    set(LIBFCL_INCLUDE_DIRS ${FCL_INCLUDE_DIRS})
+  endif()
+else()
+  find_package(PkgConfig REQUIRED)
+  pkg_check_modules(LIBFCL REQUIRED "fcl>=0.5.0")
+  # replace LIBFCL_LIBRARIES with full paths to the libraries
+  set(LIBFCL_LIBRARIES_FULL "")
+  foreach(LIBFCL_LIBRARY ${LIBFCL_LIBRARIES})
+    find_library(${LIBFCL_LIBRARY}_LIB ${LIBFCL_LIBRARY} ${LIBFCL_LIBRARY_DIRS})
+    list(APPEND LIBFCL_LIBRARIES_FULL ${${LIBFCL_LIBRARY}_LIB})
+  endforeach()
+  set(LIBFCL_LIBRARIES "${LIBFCL_LIBRARIES_FULL}")
+endif()
 
 find_package(Bullet 2.87 REQUIRED)
 find_package(angles REQUIRED)
