diff --git a/moveit_setup_assistant/moveit_setup_core_plugins/src/start_screen.cpp b/moveit_setup_assistant/moveit_setup_core_plugins/src/start_screen.cpp
index ff6827d6ff..b0550b1e90 100644
--- a/moveit_setup_assistant/moveit_setup_core_plugins/src/start_screen.cpp
+++ b/moveit_setup_assistant/moveit_setup_core_plugins/src/start_screen.cpp
@@ -70,7 +70,7 @@ std::filesystem::path StartScreen::getPackagePath()
 
 void StartScreen::loadExisting(const std::filesystem::path& package_path)
 {
-  package_settings_->loadExisting(package_path);
+  package_settings_->loadExisting(package_path.string());
 }
 
 bool StartScreen::isXacroFile()
diff --git a/moveit_setup_assistant/moveit_setup_core_plugins/src/start_screen_widget.cpp b/moveit_setup_assistant/moveit_setup_core_plugins/src/start_screen_widget.cpp
index 70bae9ddc9..82da843d60 100644
--- a/moveit_setup_assistant/moveit_setup_core_plugins/src/start_screen_widget.cpp
+++ b/moveit_setup_assistant/moveit_setup_core_plugins/src/start_screen_widget.cpp
@@ -79,7 +79,7 @@ void StartScreenWidget::onInit()
   right_image_label_ = new QLabel(this);
   auto image_path = getSharePath("moveit_setup_assistant") / "resources/MoveIt_Setup_Assistant2.png";
 
-  if (right_image_->load(image_path.c_str()))
+  if (right_image_->load(image_path.string().c_str()))
   {
     right_image_label_->setPixmap(QPixmap::fromImage(*right_image_));
     right_image_label_->setMinimumHeight(384);  // size of right_image_label_
@@ -205,7 +205,7 @@ void StartScreenWidget::focusGiven()
   std::filesystem::path pkg_path = setup_step_.getPackagePath();
   if (!pkg_path.empty())
   {
-    stack_path_->setPath(pkg_path);
+    stack_path_->setPath(pkg_path.string());
     select_mode_->btn_exist_->click();
     return;
   }
@@ -213,7 +213,7 @@ void StartScreenWidget::focusGiven()
   std::filesystem::path urdf_path = setup_step_.getURDFPath();
   if (!urdf_path.empty())
   {
-    urdf_file_->setPath(urdf_path);
+    urdf_file_->setPath(urdf_path.string());
     select_mode_->btn_new_->click();
   }
 }
diff --git a/moveit_setup_assistant/moveit_setup_core_plugins/include/moveit_setup_core_plugins/configuration_files.hpp b/moveit_setup_assistant/moveit_setup_core_plugins/include/moveit_setup_core_plugins/configuration_files.hpp
index bfca1e8d62..3d3f3ab386 100644
--- a/moveit_setup_assistant/moveit_setup_core_plugins/include/moveit_setup_core_plugins/configuration_files.hpp
+++ b/moveit_setup_assistant/moveit_setup_core_plugins/include/moveit_setup_core_plugins/configuration_files.hpp
@@ -80,7 +80,7 @@ class ConfigurationFiles : public SetupStep
 
   bool shouldGenerate(const GeneratedFilePtr& file) const
   {
-    std::string rel_path = file->getRelativePath();
+    std::string rel_path = file->getRelativePath().string();
     auto it = should_generate_.find(rel_path);
     if (it == should_generate_.end())
     {
diff --git a/moveit_setup_assistant/moveit_setup_core_plugins/src/configuration_files_widget.cpp b/moveit_setup_assistant/moveit_setup_core_plugins/src/configuration_files_widget.cpp
index 9c76fc1d52..3e689772e3 100644
--- a/moveit_setup_assistant/moveit_setup_core_plugins/src/configuration_files_widget.cpp
+++ b/moveit_setup_assistant/moveit_setup_core_plugins/src/configuration_files_widget.cpp
@@ -294,7 +294,7 @@ void ConfigurationFilesWidget::changeCheckedState(QListWidgetItem* item)
   }
 
   // Enable/disable file
-  setup_step_.setShouldGenerate(gen_file->getRelativePath(), generate);
+  setup_step_.setShouldGenerate(gen_file->getRelativePath().string(), generate);
 }
 
 // ******************************************************************************************
@@ -303,7 +303,7 @@ void ConfigurationFilesWidget::changeCheckedState(QListWidgetItem* item)
 void ConfigurationFilesWidget::focusGiven()
 {
   // Pass the package path from start screen to configuration files screen
-  stack_path_->setPath(setup_step_.getPackagePath());
+  stack_path_->setPath(setup_step_.getPackagePath().string());
 
   setup_step_.loadFiles();
 
@@ -350,7 +350,7 @@ void ConfigurationFilesWidget::showGenFiles()
     auto gen_file = gen_files[i];
 
     // Create a formatted row
-    QListWidgetItem* item = new QListWidgetItem(QString(gen_file->getRelativePath().c_str()), action_list_, 0);
+    QListWidgetItem* item = new QListWidgetItem(QString(gen_file->getRelativePath().string().c_str()), action_list_, 0);
 
     // Checkbox
     item->setCheckState(setup_step_.shouldGenerate(gen_file) ? Qt::Checked : Qt::Unchecked);
