diff --git a/CMakeLists.txt b/CMakeLists.txt
index a2ed58a..f7c7d94 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,5 +1,6 @@
 cmake_minimum_required(VERSION 3.5)
 project (popf) 
+set(CMAKE_CXX_STANDARD 14)
 
 find_package(ament_cmake REQUIRED)
 set(dependencies
@@ -16,6 +17,9 @@ find_package(CoinUtils)
 find_package(Cbc)
 find_package(Cgl)
 
+find_package(OSI REQUIRED)
+find_package(BISON REQUIRED)
+find_package(FLEX REQUIRED)
 set(val_SOURCE_DIR src/VALfiles)
 set(popf_SOURCE_DIR src/popf)
 set(parsePDDL_SOURCE_DIR ${val_SOURCE_DIR}/parsing)
@@ -29,22 +33,11 @@ include_directories(
   ${CBC_INCLUDES}/coin  
   ${CPLEX_INCLUDES} 
   ${CONCERT_INCLUDES}
+  ${FLEX_INCLUDE_DIRS}
 )
 
 ## ParsePDDL
 
-find_program(BISON_EXECUTABLE bison)
-
-if(NOT EXISTS ${BISON_EXECUTABLE})
-	message("Could not find bison executable.")
-endif(NOT EXISTS ${BISON_EXECUTABLE})
-
-FIND_PROGRAM(FLEX_EXECUTABLE flex)
-
-if(NOT EXISTS ${FLEX_EXECUTABLE})
-        message("Could not find flex executable.")
-endif(NOT EXISTS ${FLEX_EXECUTABLE})
-
 add_custom_command(
        OUTPUT
         ${CMAKE_CURRENT_BINARY_DIR}/lex.yy.cc
@@ -66,13 +59,14 @@ add_custom_command(
        DEPENDS
         ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/pddl+.yacc ${CMAKE_CURRENT_BINARY_DIR}/lex.yy.cc
        COMMAND
-        ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/fixyywrap ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/
+        ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/fixyywrap ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/${parsePDDL_SOURCE_DIR}/ ${FLEX_INCLUDE_DIRS}
 )
 
 set(libParsePDDL_SRCS
       ${CMAKE_CURRENT_BINARY_DIR}/pddl+.cpp
       ${parsePDDL_SOURCE_DIR}/ptree.cpp
       ${parsePDDL_SOURCE_DIR}/DebugWriteController.cpp
+      ${val_SOURCE_DIR}/ParserGlobals.cpp
 )
 
 add_library(ParsePDDL SHARED ${libParsePDDL_SRCS})
@@ -159,6 +153,7 @@ set(popf_build_srcs
 
 add_library(popfCommon SHARED ${popf_build_srcs})
 target_link_libraries(popfCommon ParsePDDL Inst)
+target_link_libraries(popfCommon ${CBC_LIBRARY} ${CGL_LIBRARIES} ${OSICLP_LIBRARIES} ${OSI_LIBRARIES} ${CLP_LIBRARIES} ${COINUTILS_LIBRARIES})
 
 install(TARGETS popfCommon ParsePDDL Inst
   ARCHIVE DESTINATION lib
diff --git a/src/VALfiles/ParserGlobals.cpp b/src/VALfiles/ParserGlobals.cpp
new file mode 100644
index 0000000..e1863ae
--- /dev/null
+++ b/src/VALfiles/ParserGlobals.cpp
@@ -0,0 +1,12 @@
+#include "ptree.h"
+#include "FlexLexer.h"
+
+namespace VAL
+{
+parse_category * top_thing = NULL;
+analysis an_analysis;
+analysis * current_analysis = &an_analysis;
+yyFlexLexer * yfl = NULL;
+}
+
+char * current_filename = NULL;
diff --git a/src/VALfiles/TimSupport.cpp b/src/VALfiles/TimSupport.cpp
index 9c98715..45e7d57 100644
--- a/src/VALfiles/TimSupport.cpp
+++ b/src/VALfiles/TimSupport.cpp
@@ -1552,9 +1552,11 @@ set<PropertySpace *> TIMAnalyser::relevant(pddl_type * tp)
 
 namespace std
 {
+#if !defined(_LIBCPP_VERSION)
   template<>
   struct iterator_traits<TIM::getConditionally<std::_Rb_tree_const_iterator<TIM::Property*> > >
   {
     typedef TIM::Property value_type;
   };
+#endif
 }
\ No newline at end of file
diff --git a/src/VALfiles/parse.cpp b/src/VALfiles/parse.cpp
index b51c585..5880ffe 100644
--- a/src/VALfiles/parse.cpp
+++ b/src/VALfiles/parse.cpp
@@ -44,6 +44,8 @@
 #include "ptree.h"
 #include "FlexLexer.h"
 
+//extern int yyFlexLexer::yywrap();
+//extern "C" int yywrap();
 extern int yyparse();
 extern int yydebug;
 
diff --git a/src/VALfiles/parsing/CMakeLists.txt b/src/VALfiles/parsing/CMakeLists.txt
index dd7dc8e..0627f81 100644
--- a/src/VALfiles/parsing/CMakeLists.txt
+++ b/src/VALfiles/parsing/CMakeLists.txt
@@ -2,18 +2,10 @@ cmake_minimum_required(VERSION 2.4)
 
 project(ParsePDDL)
 
-find_program(BISON_EXECUTABLE bison)
+find_package(FLEX REQUIRED)
+find_package(BISON REQUIRED)
 
-if(NOT EXISTS ${BISON_EXECUTABLE})
-	message("Could not find bison executable.")
-endif(NOT EXISTS ${BISON_EXECUTABLE})
-
-
-FIND_PROGRAM(FLEX_EXECUTABLE flex HINTS /usr/local/opt/flex/bin/)
-
-if(NOT EXISTS ${FLEX_EXECUTABLE})
-        message("Could not find flex executable.")
-endif(NOT EXISTS ${FLEX_EXECUTABLE})
+include_directories(${FLEX_INCLUDE_DIRS} ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
 
 add_custom_command(
        OUTPUT
@@ -36,11 +28,9 @@ add_custom_command(
        DEPENDS
         ${CMAKE_CURRENT_SOURCE_DIR}/pddl+.yacc ${CMAKE_CURRENT_BINARY_DIR}/lex.yy.cc
        COMMAND
-        ${CMAKE_CURRENT_SOURCE_DIR}/fixyywrap ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR}
+        ${CMAKE_CURRENT_SOURCE_DIR}/fixyywrap ${CMAKE_CURRENT_BINARY_DIR} ${CMAKE_CURRENT_SOURCE_DIR} ${FLEX_INCLUDE_DIRS}
 )
 
-include_directories(/usr/local/opt/flex/include ${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_BINARY_DIR})
-
 set(libParsePDDL_SRCS
       pddl+.cpp ptree.cpp DebugWriteController.cpp
 )
diff --git a/src/VALfiles/parsing/fixyywrap b/src/VALfiles/parsing/fixyywrap
index 11aef39..be36596 100755
--- a/src/VALfiles/parsing/fixyywrap
+++ b/src/VALfiles/parsing/fixyywrap
@@ -1,13 +1,14 @@
-#!/usr/bin/perl -w
+#!/usr/bin/env perl
 
 $bindir = shift @ARGV;
 $srcdir = shift @ARGV;
+$includepath = shift @ARGV;
 
 # MacOS' sed requires '' as a parameter to -i
-system("sed -i '' 's/yyerror(char/yyerror(const char/' $bindir/pddl+.cpp");
+system("sed 's/yyerror(char/yyerror(const char/' $bindir/pddl+.cpp");
 
 # add Homebrew's flex include path
-open IN, "g++ -E -I/usr/local/opt/flex/include $srcdir/fixyywrap.cpp |";
+open IN, "g++ -E -I$includepath $srcdir/fixyywrap.cpp |";
 open OUT, "> $bindir/fixyywrap.h";
 
 while (<IN>) {
diff --git a/src/VALfiles/parsing/pddl+.lex b/src/VALfiles/parsing/pddl+.lex
index 598c8e5..dba389c 100644
--- a/src/VALfiles/parsing/pddl+.lex
+++ b/src/VALfiles/parsing/pddl+.lex
@@ -8,7 +8,8 @@
   using std::cerr;
   using std::istream;
   using namespace VAL;
-extern "C" int yywrap();
+//extern int yyFlexLexer::yywrap();
+//extern "C" int yywrap();
 
 %}
 %option case-insensitive
diff --git a/src/VALfiles/test/lex.yy.cc b/src/VALfiles/test/lex.yy.cc
index 974fc92..cc5dfcc 100644
--- a/src/VALfiles/test/lex.yy.cc
+++ b/src/VALfiles/test/lex.yy.cc
@@ -1041,7 +1041,8 @@ static yyconst flex_int16_t yy_chk[1781] =
   using std::cerr;
   using std::istream;
   using namespace VAL;
-extern "C" int yywrap();
+//extern "C" int yywrap();
+//extern int yyFlexLexer::yywrap();
 
 #line 1047 "lex.yy.cc"
 
diff --git a/src/cmake/modules/FindOSI.cmake b/src/cmake/modules/FindOSI.cmake
index 769cd5e..52cf615 100644
--- a/src/cmake/modules/FindOSI.cmake
+++ b/src/cmake/modules/FindOSI.cmake
@@ -8,7 +8,7 @@ endif (OSI_INCLUDES AND OSI_LIBRARIES)
 
 find_path(OSI_INCLUDES
   NAMES
-  coin/config_osi.h
+  coin/OsiSolverInterface.hpp
   PATHS
   ${OSIDIR}/include
   ${INCLUDE_INSTALL_DIR}
diff --git a/src/popf/Popf.cpp b/src/popf/Popf.cpp
index adf5b5c..c25b12c 100644
--- a/src/popf/Popf.cpp
+++ b/src/popf/Popf.cpp
@@ -30,6 +30,7 @@
 #include "Popf.hpp"
 
 #include <string>
+#include <unistd.h>
 
 #include <cstdio>
 #include <iostream>
diff --git a/src/popf/RPGBuilder.h b/src/popf/RPGBuilder.h
index 851ad9a..c51d627 100644
--- a/src/popf/RPGBuilder.h
+++ b/src/popf/RPGBuilder.h
@@ -46,7 +46,6 @@ using std::map;
 #include "ptree.h"
 
 #include <assert.h>
-#include <values.h>
 #include <math.h>
 
 using namespace Inst;
diff --git a/src/popf/minimalstate.h b/src/popf/minimalstate.h
index 796b2a2..bfb8001 100644
--- a/src/popf/minimalstate.h
+++ b/src/popf/minimalstate.h
@@ -36,7 +36,6 @@
 #include <cassert>
 
 #include <float.h>  // Needed for DBL_MAX
-#include <values.h>
 
 using std::set;
 using std::map;
