diff --git a/CMakeLists.txt b/CMakeLists.txt
index a3ec429..6a64b82 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,6 +25,14 @@ ament_auto_add_library(${PROJECT_NAME}_lib SHARED
   lib/route_checker.cpp
 )
 
+if(NOT MSVC)
+  target_compile_options(${PROJECT_NAME}_lib
+    PUBLIC
+      -Wno-deprecated-declarations
+      -Wno-uninitialized
+  )
+endif()
+
 ament_auto_add_executable(${PROJECT_NAME}_sample src/sample_code.cpp)
 target_link_libraries(${PROJECT_NAME}_sample
   ${PROJECT_NAME}_lib
