diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9f8f868..bfa9cde 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,7 @@ ament_auto_add_library(pointcloud_map_loader_node SHARED
   src/pointcloud_map_loader/utils.cpp
 )
 target_link_libraries(pointcloud_map_loader_node ${PCL_LIBRARIES})
-target_link_libraries(pointcloud_map_loader_node yaml-cpp)
+target_link_libraries(pointcloud_map_loader_node yaml-cpp::yaml-cpp fmt::fmt)
 
 target_include_directories(pointcloud_map_loader_node
   SYSTEM PUBLIC
diff --git a/src/lanelet2_map_loader/lanelet2_map_loader_node.cpp b/src/lanelet2_map_loader/lanelet2_map_loader_node.cpp
index 54e9432..62ae5f6 100644
--- a/src/lanelet2_map_loader/lanelet2_map_loader_node.cpp
+++ b/src/lanelet2_map_loader/lanelet2_map_loader_node.cpp
@@ -104,7 +104,7 @@ void Lanelet2MapLoaderNode::on_map_projector_info(
     if (map_major_ver > static_cast<uint64_t>(lanelet::autoware::version)) {
       RCLCPP_WARN(
         get_logger(),
-        "format_version(%ld) of the provided map(%s) is larger than the supported version(%ld)",
+        "format_version(%llu) of the provided map(%s) is larger than the supported version(%llu)",
         map_major_ver, lanelet2_filename.c_str(),
         static_cast<uint64_t>(lanelet::autoware::version));
       if (!allow_unsupported_version) {
