diff --git a/ConfigExtras.cmake b/ConfigExtras.cmake
--- a/ConfigExtras.cmake
+++ b/ConfigExtras.cmake
@@ -1,3 +1,3 @@
 # Extras module needed for dependencies to find boost components
 
-find_package(Boost REQUIRED program_options system)
+find_package(Boost REQUIRED program_options)
diff --git a/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/NearestNeighborsGNAT.h b/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/NearestNeighborsGNAT.h
--- a/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/NearestNeighborsGNAT.h
+++ b/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/NearestNeighborsGNAT.h
@@ -39,4 +39,5 @@
 #include <moveit/exceptions/exceptions.h>
 #include <algorithm>
 #include <queue>
+#include <random>
 #include <unordered_set>
@@ -552,4 +553,4 @@
         std::vector<int> permutation(children_.size());
         for (unsigned int i = 0; i < permutation.size(); ++i)
           permutation[i] = i;
-        std::random_shuffle(permutation.begin(), permutation.end());
+        std::shuffle(permutation.begin(), permutation.end(), std::mt19937{ std::random_device{}() });
@@ -606,4 +607,4 @@
         std::vector<int> permutation(children_.size());
         for (unsigned int i = 0; i < permutation.size(); ++i)
           permutation[i] = i;
-        std::random_shuffle(permutation.begin(), permutation.end());
+        std::shuffle(permutation.begin(), permutation.end(), std::mt19937{ std::random_device{}() });
