diff --git a/CMakeLists.txt b/CMakeLists.txt
index f0d3a48..b58e32d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -484,7 +484,7 @@ set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${g2o_C_FLAGS}")
 # Find Eigen3. If it defines the target, this is used. If not,
 # fall back to the using the module form.
 # See https://eigen.tuxfamily.org/dox/TopicCMakeGuide.html for details
-find_package(Eigen3 3.3 REQUIRED)
+find_package(Eigen3 REQUIRED NO_MODULE)
 if (TARGET Eigen3::Eigen)
   set(G2O_EIGEN3_EIGEN_TARGET Eigen3::Eigen)
 else()
diff --git a/g2o/stuff/misc.h b/g2o/stuff/misc.h
index 3db52b3..5e61502 100644
--- a/g2o/stuff/misc.h
+++ b/g2o/stuff/misc.h
@@ -28,6 +28,7 @@
 #ifndef G2O_STUFF_MISC_H
 #define G2O_STUFF_MISC_H
 
+#include <cassert>
 #include <cmath>
 #include <memory>
 
