diff --git a/CMakeLists.txt b/CMakeLists.txt
index b74b6dd..62db05f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -117,6 +117,7 @@ target_include_directories(rviz_rendering
     $<INSTALL_INTERFACE:include/${PROJECT_NAME}>
     ${Eigen3_INCLUDE_DIRS}
 )
+target_link_libraries(rviz_rendering PUBLIC Eigen3::Eigen)
 # Causes the visibility macros to use dllexport rather than dllimport,
 # which is appropriate when building the dll but not consuming it.
 target_compile_definitions(rviz_rendering PRIVATE "RVIZ_RENDERING_BUILDING_LIBRARY")
