diff --git a/diff_drive_controller/include/diff_drive_controller/diff_drive_controller.hpp b/diff_drive_controller/include/diff_drive_controller/diff_drive_controller.hpp
index 72b38f7d2d..1ec3ef5427 100644
--- a/diff_drive_controller/include/diff_drive_controller/diff_drive_controller.hpp
+++ b/diff_drive_controller/include/diff_drive_controller/diff_drive_controller.hpp
@@ -98,8 +98,9 @@ class DiffDriveController : public controller_interface::ControllerInterface
     std::reference_wrapper<const hardware_interface::LoanedStateInterface> feedback;
     std::reference_wrapper<hardware_interface::LoanedCommandInterface> velocity;
   };
-
+  DIFF_DRIVE_CONTROLLER_PUBLIC
   const char * feedback_type() const;
+  DIFF_DRIVE_CONTROLLER_PUBLIC
   controller_interface::CallbackReturn configure_side(
     const std::string & side, const std::vector<std::string> & wheel_names,
     std::vector<WheelHandle> & registered_handles);
@@ -155,8 +156,9 @@ class DiffDriveController : public controller_interface::ControllerInterface
   rclcpp::Time previous_publish_timestamp_{0, 0, RCL_CLOCK_UNINITIALIZED};
 
   bool is_halted = false;
-
+  DIFF_DRIVE_CONTROLLER_PUBLIC
   bool reset();
+  DIFF_DRIVE_CONTROLLER_PUBLIC
   void halt();
 };
 }  // namespace diff_drive_controller
