diff --git a/ConfigExtras.cmake b/ConfigExtras.cmake
--- a/ConfigExtras.cmake
+++ b/ConfigExtras.cmake
@@ -1,3 +1,3 @@
 # Extras module needed for dependencies to find boost components
 
-find_package(Boost REQUIRED program_options system)
+find_package(Boost REQUIRED program_options)
diff --git a/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/NearestNeighborsGNAT.h b/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/NearestNeighborsGNAT.h
--- a/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/NearestNeighborsGNAT.h
+++ b/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/NearestNeighborsGNAT.h
@@ -39,4 +39,5 @@
 #include <moveit/exceptions/exceptions.h>
 #include <algorithm>
 #include <queue>
+#include <random>
 #include <unordered_set>
@@ -552,4 +553,4 @@
         std::vector<int> permutation(children_.size());
         for (unsigned int i = 0; i < permutation.size(); ++i)
           permutation[i] = i;
-        std::random_shuffle(permutation.begin(), permutation.end());
+        std::shuffle(permutation.begin(), permutation.end(), std::mt19937{ std::random_device{}() });
@@ -606,4 +607,4 @@
         std::vector<int> permutation(children_.size());
         for (unsigned int i = 0; i < permutation.size(); ++i)
           permutation[i] = i;
-        std::random_shuffle(permutation.begin(), permutation.end());
+        std::shuffle(permutation.begin(), permutation.end(), std::mt19937{ std::random_device{}() });
diff --git a/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/GreedyKCenters.h b/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/GreedyKCenters.h
--- a/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/GreedyKCenters.h
+++ b/cached_ik_kinematics_plugin/include/moveit/cached_ik_kinematics_plugin/detail/GreedyKCenters.h
@@ -92,7 +92,7 @@
     centers.clear();
     centers.reserve(k);
     if (((long unsigned int)dists.rows()) < data.size() || ((long unsigned int)dists.cols()) < k)
-      dists.resize(std::max(2 * ((long unsigned int)dists.rows()) + 1, data.size()), k);
+      dists.resize(std::max<size_t>(2 * static_cast<size_t>(dists.rows()) + 1, data.size()), k);
     // first center is picked randomly
     centers.push_back(std::uniform_int_distribution<size_t>{ 0, data.size() - 1 }(generator_));
     for (unsigned i = 1; i < k; ++i)
