diff --git a/franka_robot_state_broadcaster/CMakeLists.txt b/franka_robot_state_broadcaster/CMakeLists.txt
index 4a71b0c1..c1437ebe 100644
--- a/franka_robot_state_broadcaster/CMakeLists.txt
+++ b/franka_robot_state_broadcaster/CMakeLists.txt
@@ -97,7 +97,7 @@ if(BUILD_TESTING)
         ${CMAKE_CURRENT_SOURCE_DIR}/test/franka_robot_state_broadcaster_parameters.yaml)
     target_include_directories(${PROJECT_NAME}_test PRIVATE include)
     target_link_libraries(${PROJECT_NAME}_test ${PROJECT_NAME})
-    ament_target_dependencies(${PROJECT_NAME}_test controller_manager hardware_interface)
+    ament_target_dependencies(${PROJECT_NAME}_test controller_manager hardware_interface ros2_control_test_assets)
 
     ament_add_gmock(test_load_franka_robot_state_broadcaster test/test_load_franka_robot_state_broadcaster.cpp)
     target_include_directories(test_load_franka_robot_state_broadcaster PRIVATE include)

