diff --git a/rosbag2_storage/CMakeLists.txt b/rosbag2_storage/CMakeLists.txt
index d31ad8de3..8a17fc870 100644
--- a/rosbag2_storage/CMakeLists.txt
+++ b/rosbag2_storage/CMakeLists.txt
@@ -39,12 +39,13 @@ target_include_directories(${PROJECT_NAME}
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include/${PROJECT_NAME}>
 )
-ament_target_dependencies(
-  ${PROJECT_NAME}
-  pluginlib
-  rcpputils
-  rcutils
-  yaml_cpp_vendor)
+target_link_libraries(${PROJECT_NAME}
+  pluginlib::pluginlib
+  rcutils::rcutils
+  rcpputils::rcpputils
+  
+  yaml-cpp::yaml-cpp
+)
 
 # Causes the visibility macros to use dllexport rather than dllimport,
 # which is appropriate when building the dll but not consuming it.
