diff --git a/include/grid_map_core/eigen_plugins/FunctorsPlugin.hpp b/include/grid_map_core/eigen_plugins/FunctorsPlugin.hpp
index b5c3abb..08aa30b 100644
--- a/include/grid_map_core/eigen_plugins/FunctorsPlugin.hpp
+++ b/include/grid_map_core/eigen_plugins/FunctorsPlugin.hpp
@@ -6,7 +6,7 @@
 template<typename Scalar>
 struct scalar_sum_of_finites_op
 {
-  EIGEN_EMPTY_STRUCT_CTOR(scalar_sum_of_finites_op)
+  scalar_sum_of_finites_op() = default;
   EIGEN_STRONG_INLINE const Scalar operator()(const Scalar & a, const Scalar & b) const
   {
     using std::isfinite;
@@ -29,7 +29,7 @@ struct functor_traits<scalar_sum_of_finites_op<Scalar>>
 template<typename Scalar>
 struct scalar_min_of_finites_op
 {
-  EIGEN_EMPTY_STRUCT_CTOR(scalar_min_of_finites_op)
+  scalar_min_of_finites_op() = default;
   EIGEN_STRONG_INLINE const Scalar operator()(const Scalar & a, const Scalar & b) const
   {
     using std::min;
@@ -53,7 +53,7 @@ struct functor_traits<scalar_min_of_finites_op<Scalar>>
 template<typename Scalar>
 struct scalar_max_of_finites_op
 {
-  EIGEN_EMPTY_STRUCT_CTOR(scalar_max_of_finites_op)
+  scalar_max_of_finites_op() = default;
   EIGEN_STRONG_INLINE const Scalar operator()(const Scalar & a, const Scalar & b) const
   {
     using std::max;
