diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3add4d5..c0ce5bb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,7 +70,7 @@ function(configure_build_install_location _library_name)
 endfunction()
 
 # Split from main extension and converted to pybind11
-pybind11_add_module(_rclpy_pybind11
+pybind11_add_module(_rclpy_pybind11 MODULE NO_EXTRAS
   src/rclpy/_rclpy_logging.cpp
   src/rclpy/_rclpy_pybind11.cpp
   src/rclpy/action_client.cpp
diff --git a/src/rclpy/_rclpy_logging.cpp b/src/rclpy/_rclpy_logging.cpp
index c510b0a..854e79e 100644
--- a/src/rclpy/_rclpy_logging.cpp
+++ b/src/rclpy/_rclpy_logging.cpp
@@ -129,7 +129,7 @@ rclpy_logging_rcutils_log(
   uint64_t line_number)
 {
   RCUTILS_LOGGING_AUTOINIT;
-  rcutils_log_location_t logging_location = {function_name, file_name, line_number};
+  rcutils_log_location_t logging_location = {function_name, file_name,  static_cast<size_t>(line_number)};
   rcutils_log(&logging_location, severity, name, "%s", message);
 }
 
