diff --git a/pid_controller/CMakeLists.txt b/pid_controller/CMakeLists.txt
index 81cbe6f006..d7798b579b 100644
--- a/pid_controller/CMakeLists.txt
+++ b/pid_controller/CMakeLists.txt
@@ -45,7 +45,7 @@ ament_target_dependencies(pid_controller PUBLIC ${THIS_PACKAGE_INCLUDE_DEPENDS})
 
 # Causes the visibility macros to use dllexport rather than dllimport,
 # which is appropriate when building the dll but not consuming it.
-target_compile_definitions(pid_controller PRIVATE "PID_CONTROLLER_BUILDING_DLL")
+target_compile_definitions(pid_controller PRIVATE "PID_CONTROLLER__VISIBILITY_BUILDING_DLL")
 
 pluginlib_export_plugin_description_file(controller_interface pid_controller.xml)
 
diff --git a/pid_controller/include/pid_controller/pid_controller.hpp b/pid_controller/include/pid_controller/pid_controller.hpp
index 105a915442..4919a739ce 100644
--- a/pid_controller/include/pid_controller/pid_controller.hpp
+++ b/pid_controller/include/pid_controller/pid_controller.hpp
@@ -119,8 +119,10 @@ class PidController : public controller_interface::ChainableControllerInterface
   std::unique_ptr<ControllerStatePublisher> state_publisher_;
 
   // override methods from ChainableControllerInterface
+  PID_CONTROLLER__VISIBILITY_PUBLIC
   std::vector<hardware_interface::CommandInterface> on_export_reference_interfaces() override;
 
+  PID_CONTROLLER__VISIBILITY_PUBLIC
   bool on_set_chained_mode(bool chained_mode) override;
 
   // internal methods
