diff --git a/CMakeLists.txt b/CMakeLists.txt
index 504411d57..a11cd4ce6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,24 +13,25 @@ endif()
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(std_msgs REQUIRED)
+find_package(Python COMPONENTS Interpreter Development REQUIRED)
 
 add_executable(publisher_lambda lambda.cpp)
-ament_target_dependencies(publisher_lambda rclcpp std_msgs)
+target_link_libraries(publisher_lambda PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
 
 add_executable(publisher_member_function member_function.cpp)
-ament_target_dependencies(publisher_member_function rclcpp std_msgs)
+target_link_libraries(publisher_member_function PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
 
 add_executable(publisher_member_function_with_type_adapter member_function_with_type_adapter.cpp)
-ament_target_dependencies(publisher_member_function_with_type_adapter rclcpp std_msgs)
+target_link_libraries(publisher_member_function_with_type_adapter PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
 
 add_executable(publisher_member_function_with_unique_network_flow_endpoints member_function_with_unique_network_flow_endpoints.cpp)
-ament_target_dependencies(publisher_member_function_with_unique_network_flow_endpoints rclcpp std_msgs)
+target_link_libraries(publisher_member_function_with_unique_network_flow_endpoints PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
 
 add_executable(publisher_wait_for_all_acked member_function_with_wait_for_all_acked.cpp)
-ament_target_dependencies(publisher_wait_for_all_acked rclcpp std_msgs)
+target_link_libraries(publisher_wait_for_all_acked PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
 
 add_executable(publisher_not_composable not_composable.cpp)
-ament_target_dependencies(publisher_not_composable rclcpp std_msgs)
+target_link_libraries(publisher_not_composable PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
 
 install(TARGETS
   publisher_lambda
