diff --git a/include/mavconn/serial.hpp b/include/mavconn/serial.hpp
index 3fc232db7..7152692cc 100644
--- a/include/mavconn/serial.hpp
+++ b/include/mavconn/serial.hpp
@@ -66,7 +66,7 @@ public:
   }
 
 private:
-  asio::io_service io_service;
+  asio::io_context io_service;
   std::thread io_thread;
   asio::serial_port serial_dev;
 
diff --git a/include/mavconn/tcp.hpp b/include/mavconn/tcp.hpp
index ba0064191..5ff038ea1 100644
--- a/include/mavconn/tcp.hpp
+++ b/include/mavconn/tcp.hpp
@@ -60,7 +60,7 @@ public:
    */
   explicit MAVConnTCPClient(
     uint8_t system_id, uint8_t component_id,
-    asio::io_service & server_io);
+    asio::io_context & server_io);
 
   virtual ~MAVConnTCPClient();
 
@@ -80,8 +80,8 @@ public:
 
 private:
   friend class MAVConnTCPServer;
-  asio::io_service io_service;
-  std::unique_ptr<asio::io_service::work> io_work;
+  asio::io_context io_service;
+  std::unique_ptr<asio::executor_work_guard<asio::io_context::executor_type>> io_work;
   std::thread io_thread;
   std::atomic<bool> is_running;  //!< io_thread running
 
@@ -147,8 +147,8 @@ public:
   }
 
 private:
-  asio::io_service io_service;
-  std::unique_ptr<asio::io_service::work> io_work;
+  asio::io_context io_service;
+  std::unique_ptr<asio::executor_work_guard<asio::io_context::executor_type>> io_work;
   std::thread io_thread;
 
   asio::ip::tcp::acceptor acceptor;
diff --git a/include/mavconn/udp.hpp b/include/mavconn/udp.hpp
index 66266404c..bd5d9a267 100644
--- a/include/mavconn/udp.hpp
+++ b/include/mavconn/udp.hpp
@@ -79,8 +79,8 @@ public:
   std::string get_remote_endpoint() const;
 
 private:
-  asio::io_service io_service;
-  std::unique_ptr<asio::io_service::work> io_work;
+  asio::io_context io_service;
+  std::unique_ptr<asio::executor_work_guard<asio::io_context::executor_type>> io_work;
   std::thread io_thread;
   std::atomic<bool> is_running;  //!< io_thread running
   bool permanent_broadcast;
diff --git a/src/serial.cpp b/src/serial.cpp
index 690bebde5..6a1455ce7 100644
--- a/src/serial.cpp
+++ b/src/serial.cpp
@@ -30,7 +30,6 @@ namespace mavconn
 {
 
 using asio::buffer;
-using asio::io_service;
 using mavlink::mavlink_message_t;
 using std::error_code;
 
@@ -123,7 +122,7 @@ void MAVConnSerial::connect(
   port_closed_cb = cb_handle_closed_port;
 
   // give some work to io_service before start
-  io_service.post(std::bind(&MAVConnSerial::do_read, this));
+  asio::post(io_service, std::bind(&MAVConnSerial::do_read, this));
 
   // run io_service for async io
   io_thread = std::thread(
@@ -150,7 +149,7 @@ void MAVConnSerial::close()
     io_thread.join();
   }
 
-  io_service.reset();
+  io_service.restart();
 
   if (port_closed_cb) {
     port_closed_cb();
@@ -173,7 +172,7 @@ void MAVConnSerial::send_bytes(const uint8_t * bytes, size_t length)
 
     tx_q.emplace_back(bytes, length);
   }
-  io_service.post(std::bind(&MAVConnSerial::do_write, shared_from_this(), true));
+  asio::post(io_service, std::bind(&MAVConnSerial::do_write, shared_from_this(), true));
 }
 
 void MAVConnSerial::send_message(const mavlink_message_t * message)
@@ -196,7 +195,7 @@ void MAVConnSerial::send_message(const mavlink_message_t * message)
 
     tx_q.emplace_back(message);
   }
-  io_service.post(std::bind(&MAVConnSerial::do_write, shared_from_this(), true));
+  asio::post(io_service, std::bind(&MAVConnSerial::do_write, shared_from_this(), true));
 }
 
 void MAVConnSerial::send_message(const mavlink::Message & message, const uint8_t source_compid)
@@ -217,7 +216,7 @@ void MAVConnSerial::send_message(const mavlink::Message & message, const uint8_t
 
     tx_q.emplace_back(message, get_status_p(), sys_id, source_compid);
   }
-  io_service.post(std::bind(&MAVConnSerial::do_write, shared_from_this(), true));
+  asio::post(io_service, std::bind(&MAVConnSerial::do_write, shared_from_this(), true));
 }
 
 void MAVConnSerial::do_read(void)
diff --git a/src/tcp.cpp b/src/tcp.cpp
index e5d6585e3..33f9053c6 100644
--- a/src/tcp.cpp
+++ b/src/tcp.cpp
@@ -24,19 +24,14 @@
 #include "mavconn/tcp.hpp"
 #include "mavconn/thread_utils.hpp"
 
-// Ensure the correct io_service() is called based on asio version
-#if ASIO_VERSION >= 101400
 #define GET_IO_SERVICE(s) ((asio::io_context &)(s).get_executor().context())
-#else
-#define GET_IO_SERVICE(s) ((s).get_io_service())
-#endif
 
 namespace mavconn
 {
 
 using asio::buffer;
 using asio::error_code;
-using asio::io_service;
+using asio::io_context;
 using asio::ip::tcp;
 using mavlink::mavlink_message_t;
 using mavlink::mavlink_status_t;
@@ -46,15 +41,13 @@ using utils::to_string_ss;
 #define PFXd PFX "%zu: "
 
 static bool resolve_address_tcp(
-  io_service & io, size_t chan, std::string host, uint16_t port,
+  io_context & io, size_t chan, std::string host, uint16_t port,
   tcp::endpoint & ep)
 {
   bool result = false;
   tcp::resolver resolver(io);
   error_code ec;
 
-  tcp::resolver::query query(host, "");
-
   auto fn = [&](const tcp::endpoint & q_ep) {
       ep = q_ep;
       ep.port(port);
@@ -64,11 +57,11 @@ static bool resolve_address_tcp(
     };
 
 #if ASIO_VERSION >= 101200
-  for (auto q_ep : resolver.resolve(query, ec)) {
+  for (auto q_ep : resolver.resolve(host, "", ec)) {
     fn(q_ep);
   }
 #else
-  std::for_each(resolver.resolve(query, ec), tcp::resolver::iterator(), fn);
+  std::for_each(resolver.resolve(host, "", ec), tcp::resolver::iterator(), fn);
 #endif
 
   if (ec) {
@@ -86,7 +79,8 @@ MAVConnTCPClient::MAVConnTCPClient(
   std::string server_host, uint16_t server_port)
 : MAVConnInterface(system_id, component_id),
   io_service(),
-  io_work(new io_service::work(io_service)),
+  io_work(std::make_unique<asio::executor_work_guard<asio::io_context::executor_type>>(
+      asio::make_work_guard(io_service))),
   is_running(false),
   socket(io_service),
   is_destroying(false),
@@ -110,7 +104,7 @@ MAVConnTCPClient::MAVConnTCPClient(
 
 MAVConnTCPClient::MAVConnTCPClient(
   uint8_t system_id, uint8_t component_id,
-  asio::io_service & server_io)
+  asio::io_context & server_io)
 : MAVConnInterface(system_id, component_id),
   is_running(false),
   socket(server_io),
@@ -129,7 +123,7 @@ void MAVConnTCPClient::client_connected(size_t server_channel)
     server_channel, conn_id, to_string_ss(server_ep).c_str());
 
   // start recv
-  GET_IO_SERVICE(socket).post(std::bind(&MAVConnTCPClient::do_recv, shared_from_this()));
+  asio::post(GET_IO_SERVICE(socket), std::bind(&MAVConnTCPClient::do_recv, shared_from_this()));
 }
 
 MAVConnTCPClient::~MAVConnTCPClient()
@@ -152,7 +146,7 @@ void MAVConnTCPClient::connect(
   port_closed_cb = cb_handle_closed_port;
 
   // give some work to io_service before start
-  io_service.post(std::bind(&MAVConnTCPClient::do_recv, this));
+  asio::post(io_service, std::bind(&MAVConnTCPClient::do_recv, this));
 
   // run io_service for async io
   io_thread = std::thread(
@@ -177,7 +171,7 @@ void MAVConnTCPClient::stop()
     io_thread.join();
   }
 
-  io_service.reset();
+  io_service.restart();
 }
 
 void MAVConnTCPClient::close()
@@ -223,7 +217,8 @@ void MAVConnTCPClient::send_bytes(const uint8_t * bytes, size_t length)
 
     tx_q.emplace_back(bytes, length);
   }
-  GET_IO_SERVICE(socket).post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
+  asio::post(
+    GET_IO_SERVICE(socket), std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
 }
 
 void MAVConnTCPClient::send_message(const mavlink_message_t * message)
@@ -246,7 +241,8 @@ void MAVConnTCPClient::send_message(const mavlink_message_t * message)
 
     tx_q.emplace_back(message);
   }
-  GET_IO_SERVICE(socket).post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
+  asio::post(
+    GET_IO_SERVICE(socket), std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
 }
 
 void MAVConnTCPClient::send_message(const mavlink::Message & message, const uint8_t source_compid)
@@ -267,7 +263,8 @@ void MAVConnTCPClient::send_message(const mavlink::Message & message, const uint
 
     tx_q.emplace_back(message, get_status_p(), sys_id, source_compid);
   }
-  GET_IO_SERVICE(socket).post(std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
+  asio::post(
+    GET_IO_SERVICE(socket), std::bind(&MAVConnTCPClient::do_send, shared_from_this(), true));
 }
 
 void MAVConnTCPClient::do_recv()
@@ -376,7 +373,7 @@ void MAVConnTCPServer::connect(
   port_closed_cb = cb_handle_closed_port;
 
   // give some work to io_service before start
-  io_service.post(std::bind(&MAVConnTCPServer::do_accept, this));
+  asio::post(io_service, std::bind(&MAVConnTCPServer::do_accept, this));
 
   // run io_service for async io
   io_thread = std::thread(
diff --git a/src/udp.cpp b/src/udp.cpp
index 2d513a3bd..e3f9bee27 100644
--- a/src/udp.cpp
+++ b/src/udp.cpp
@@ -27,7 +27,7 @@ namespace mavconn
 
 using asio::buffer;
 using asio::error_code;
-using asio::io_service;
+using asio::io_context;
 using asio::ip::udp;
 using utils::to_string_ss;
 using utils::operator"" _KiB;
@@ -37,15 +37,13 @@ using mavlink::mavlink_message_t;
 #define PFXd PFX "%zu: "
 
 static bool resolve_address_udp(
-  io_service & io, size_t chan, std::string host, unsigned short port,
+  io_context & io, size_t chan, std::string host, unsigned short port,
   udp::endpoint & ep)
 {
   bool result = false;
   udp::resolver resolver(io);
   error_code ec;
 
-  udp::resolver::query query(host, "");
-
   auto fn = [&](const udp::endpoint & q_ep) {
       ep = q_ep;
       ep.port(port);
@@ -55,11 +53,11 @@ static bool resolve_address_udp(
     };
 
 #if ASIO_VERSION >= 101200
-  for (auto q_ep : resolver.resolve(query, ec)) {
+  for (auto q_ep : resolver.resolve(host, "", ec)) {
     fn(q_ep);
   }
 #else
-  std::for_each(resolver.resolve(query, ec), udp::resolver::iterator(), fn);
+  std::for_each(resolver.resolve(host, "", ec), udp::resolver::iterator(), fn);
 #endif
 
   if (ec) {
@@ -76,7 +74,8 @@ MAVConnUDP::MAVConnUDP(
   std::string remote_host, uint16_t remote_port)
 : MAVConnInterface(system_id, component_id),
   io_service(),
-  io_work(new io_service::work(io_service)),
+  io_work(std::make_unique<asio::executor_work_guard<asio::io_context::executor_type>>(
+      asio::make_work_guard(io_service))),
   is_running(false),
   permanent_broadcast(false),
   remote_exists(false),
@@ -147,7 +146,7 @@ void MAVConnUDP::connect(
   port_closed_cb = cb_handle_closed_port;
 
   // give some work to io_service before start
-  io_service.post(std::bind(&MAVConnUDP::do_recvfrom, this));
+  asio::post(io_service, std::bind(&MAVConnUDP::do_recvfrom, this));
 
   // run io_service for async io
   io_thread = std::thread(
@@ -172,7 +171,7 @@ void MAVConnUDP::stop()
     io_thread.join();
   }
 
-  io_service.reset();
+  io_service.restart();
 }
 
 void MAVConnUDP::close()
@@ -218,7 +217,7 @@ void MAVConnUDP::send_bytes(const uint8_t * bytes, size_t length)
 
     tx_q.emplace_back(bytes, length);
   }
-  io_service.post(std::bind(&MAVConnUDP::do_sendto, shared_from_this(), true));
+  asio::post(io_service, std::bind(&MAVConnUDP::do_sendto, shared_from_this(), true));
 }
 
 void MAVConnUDP::send_message(const mavlink_message_t * message)
@@ -246,7 +245,7 @@ void MAVConnUDP::send_message(const mavlink_message_t * message)
 
     tx_q.emplace_back(message);
   }
-  io_service.post(std::bind(&MAVConnUDP::do_sendto, shared_from_this(), true));
+  asio::post(io_service, std::bind(&MAVConnUDP::do_sendto, shared_from_this(), true));
 }
 
 void MAVConnUDP::send_message(const mavlink::Message & message, const uint8_t source_compid)
@@ -272,7 +271,7 @@ void MAVConnUDP::send_message(const mavlink::Message & message, const uint8_t so
 
     tx_q.emplace_back(message, get_status_p(), sys_id, source_compid);
   }
-  io_service.post(std::bind(&MAVConnUDP::do_sendto, shared_from_this(), true));
+  asio::post(io_service, std::bind(&MAVConnUDP::do_sendto, shared_from_this(), true));
 }
 
 void MAVConnUDP::do_recvfrom()

