diff --git a/CMakeLists.txt b/CMakeLists.txt
index 569a898..8900632 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -70,7 +72,7 @@ function(configure_build_install_location _library_name)
 endfunction()
 
 # Split from main extension and converted to pybind11
-pybind11_add_module(_rclpy_pybind11 SHARED
+pybind11_add_module(_rclpy_pybind11 MODULE NO_EXTRAS
   src/rclpy/_rclpy_logging.cpp
   src/rclpy/_rclpy_pybind11.cpp
   src/rclpy/action_client.cpp
@@ -116,6 +116,9 @@ target_link_libraries(_rclpy_pybind11 PRIVATE
   rcutils::rcutils
   rosidl_runtime_c::rosidl_runtime_c
 )
+  set_target_properties(_rclpy_pybind11 PROPERTIES
+    LINK_FLAGS "-sWASM_BIGINT=1 -sSIDE_MODULE=1 -sLINKABLE=1 -sEXPORT_ALL=1 -sMODULARIZE=1"
+  )
 configure_build_install_location(_rclpy_pybind11)
 
 if(NOT WIN32)
diff --git a/src/rclpy/_rclpy_logging.cpp b/src/rclpy/_rclpy_logging.cpp
index c510b0a..854e79e 100644
--- a/src/rclpy/_rclpy_logging.cpp
+++ b/src/rclpy/_rclpy_logging.cpp
@@ -129,7 +129,7 @@ rclpy_logging_rcutils_log(
   uint64_t line_number)
 {
   RCUTILS_LOGGING_AUTOINIT;
-  rcutils_log_location_t logging_location = {function_name, file_name, line_number};
+  rcutils_log_location_t logging_location = {function_name, file_name,  static_cast<size_t>(line_number)};
   rcutils_log(&logging_location, severity, name, "%s", message);
 }
 
