diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b9cda78..5fe961e9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,7 @@ find_package(class_loader REQUIRED)
 find_package(composition_interfaces REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rcpputils REQUIRED)
+find_package(Python COMPONENTS Interpreter Development REQUIRED)
 
 # Add an interface library that can be dependend upon by libraries who register components
 add_library(component INTERFACE)
@@ -37,12 +38,15 @@ add_library(
 target_include_directories(component_manager PUBLIC
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
   "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
-ament_target_dependencies(component_manager
-  "ament_index_cpp"
-  "class_loader"
-  "composition_interfaces"
-  "rclcpp"
-  "rcpputils"
+target_link_libraries(component_manager PUBLIC
+  ${composition_interfaces_TARGETS}
+  rclcpp::rclcpp
+)
+target_link_libraries(component_manager PRIVATE
+  ament_index_cpp::ament_index_cpp
+  class_loader::class_loader
+  rcpputils::rcpputils
+  Python::Python
 )
 target_compile_definitions(component_manager
   PRIVATE "RCLCPP_COMPONENTS_BUILDING_LIBRARY")
@@ -51,10 +55,7 @@ add_executable(
   component_container
   src/component_container.cpp
 )
-target_link_libraries(component_container component_manager)
-ament_target_dependencies(component_container
-  "rclcpp"
-)
+target_link_libraries(component_container component_manager rclcpp::rclcpp)
 
 set(node_main_template_install_dir "share/${PROJECT_NAME}")
 install(FILES
@@ -65,19 +66,13 @@ add_executable(
   component_container_mt
   src/component_container_mt.cpp
 )
-target_link_libraries(component_container_mt component_manager)
-ament_target_dependencies(component_container_mt
-  "rclcpp"
-)
+target_link_libraries(component_container_mt component_manager rclcpp::rclcpp)
 
 add_executable(
   component_container_isolated
   src/component_container_isolated.cpp
 )
-target_link_libraries(component_container_isolated component_manager)
-ament_target_dependencies(component_container_isolated
-  "rclcpp"
-)
+target_link_libraries(component_container_isolated component_manager rclcpp::rclcpp)
 
 
 if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
