diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8927192..24131aa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -108,7 +108,7 @@ endif()
 
 install(TARGETS backward
   DESTINATION lib)
-ament_export_libraries(backward)
+# ament_export_libraries(backward)
 
 install(DIRECTORY include/
   DESTINATION include)
diff --git a/cmake/BackwardConfig.cmake b/cmake/BackwardConfig.cmake
index 77d22d2..38b58cf 100644
--- a/cmake/BackwardConfig.cmake
+++ b/cmake/BackwardConfig.cmake
@@ -123,6 +123,10 @@ foreach(def ${BACKWARD_DEFINITIONS})
 	message(STATUS "${def}")
 endforeach()
 
+if(WIN32)
+	list(APPEND _BACKWARD_LIBRARIES dbghelp psapi)
+endif()
+
 set(BACKWARD_INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}")
 
 include(FindPackageHandleStandardArgs)
