diff --git a/hardware_interface/include/hardware_interface/lexical_casts.hpp b/hardware_interface/include/hardware_interface/lexical_casts.hpp
index 846d9f757c..042361e392 100644
--- a/hardware_interface/include/hardware_interface/lexical_casts.hpp
+++ b/hardware_interface/include/hardware_interface/lexical_casts.hpp
@@ -21,6 +21,8 @@
 #include <stdexcept>
 #include <string>
 
+#include "hardware_interface/visibility_control.h"
+
 namespace hardware_interface
 {
 
@@ -29,8 +31,10 @@ namespace hardware_interface
  * from
  https://github.com/ros-planning/srdfdom/blob/ad17b8d25812f752c397a6011cec64aeff090c46/src/model.cpp#L53
 */
+HARDWARE_INTERFACE_PUBLIC
 double stod(const std::string & s);
 
+HARDWARE_INTERFACE_PUBLIC
 bool parse_bool(const std::string & bool_string);
 
 }  // namespace hardware_interface
