diff --git a/joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp b/joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp
index f1c532dce9..9eb7ab8a13 100644
--- a/joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp
+++ b/joint_state_broadcaster/include/joint_state_broadcaster/joint_state_broadcaster.hpp
@@ -88,9 +88,13 @@ class JointStateBroadcaster : public controller_interface::ControllerInterface
     const rclcpp_lifecycle::State & previous_state) override;
 
 protected:
+  JOINT_STATE_BROADCASTER_PUBLIC
   bool init_joint_data();
+  JOINT_STATE_BROADCASTER_PUBLIC
   void init_joint_state_msg();
+  JOINT_STATE_BROADCASTER_PUBLIC
   void init_dynamic_joint_state_msg();
+  JOINT_STATE_BROADCASTER_PUBLIC
   bool use_all_available_interfaces() const;
 
 protected:
