diff --git a/kinematics_interface/CMakeLists.txt b/kinematics_interface/CMakeLists.txt
index 27a6269..2a1a3ef 100644
--- a/kinematics_interface/CMakeLists.txt
+++ b/kinematics_interface/CMakeLists.txt
@@ -5,6 +5,10 @@ if(CMAKE_CXX_COMPILER_ID MATCHES "(GNU|Clang)")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
 
+# using this instead of visibility macros
+# S1 from https://github.com/ros-controls/ros2_controllers/issues/1053
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+
 set(THIS_PACKAGE_INCLUDE_DEPENDS
   rclcpp
   rclcpp_lifecycle
