diff --git a/CMakeLists.txt b/CMakeLists.txt
index c6aa1cb..0b8dc88 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -39,9 +39,15 @@ target_include_directories(${PROJECT_NAME} PUBLIC
 target_link_libraries(${PROJECT_NAME} PUBLIC
   rosidl_runtime_c::rosidl_runtime_c
   rosidl_typesupport_c::rosidl_typesupport_c)
-target_link_libraries(${PROJECT_NAME} PRIVATE
-  rcpputils::rcpputils
-  rcutils::rcutils)
+
+if (EMSCRIPTEN)
+  target_link_libraries(${PROJECT_NAME} PRIVATE
+    rcpputils::rcpputils)
+else()
+  target_link_libraries(${PROJECT_NAME} PRIVATE
+    rcpputils::rcpputils
+    rcutils::rcutils)
+endif()
 
 # Export old-style CMake variables
 ament_export_include_directories("include/${PROJECT_NAME}")

