diff --git a/CMakeLists.txt b/CMakeLists.txt
index b686167..80a6ae8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,8 +30,8 @@ endif()
 # find *absolute* paths to ASSIMP_LIBRARIES
 # Both, pkg-config and assimp's cmake-config don't provide an absolute library path.
 # For, pkg-config the path is in ASSIMP_PC_LIBRARY_DIRS, for cmake in ASSIMP_LIBRARY_DIRS.
-find_library(assimp_ABS_LIBRARIES NAMES ${assimp_LIBRARIES} assimp HINTS ${assimp_LIBRARY_DIRS} ${assimp_PC_LIBRARY_DIRS})
-set(assimp_LIBRARIES "${assimp_ABS_LIBRARIES}")
+# find_library(assimp_ABS_LIBRARIES NAMES ${assimp_LIBRARIES} assimp HINTS ${assimp_LIBRARY_DIRS} ${assimp_PC_LIBRARY_DIRS})
+# set(assimp_LIBRARIES "${assimp_ABS_LIBRARIES}")
 
 # These need to be in this order to find header files
 find_package(eigen3_cmake_module REQUIRED)
@@ -90,10 +90,12 @@ ament_target_dependencies(${PROJECT_NAME}
 )
 ament_target_dependencies(${PROJECT_NAME} SYSTEM
   # We don't export these dependencies because their cmake is broken
-  assimp
   QHULL
 )
 
+get_property(ASSIMP_LIBS TARGET assimp::assimp PROPERTY IMPORTED_IMPLIB_RELEASE)
+target_link_libraries(${PROJECT_NAME} ${ASSIMP_LIBS})
+
 if(BUILD_TESTING)
   find_package(ament_cmake_gtest REQUIRED)
   # Unit tests
