From 4ebd2c68ab3eebb141c30a7ccf1dd6f9627328c1 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Mon, 12 Feb 2024 13:35:50 +0100
Subject: [PATCH] Add missing export symbol in protected methods

---
 .../controller_interface/chainable_controller_interface.hpp      | 1 +
 1 file changed, 1 insertion(+)

diff --git a/controller_interface/include/controller_interface/chainable_controller_interface.hpp b/controller_interface/include/controller_interface/chainable_controller_interface.hpp
index 2bdccefdc5..13c8b3f9b9 100644
--- a/controller_interface/include/controller_interface/chainable_controller_interface.hpp
+++ b/controller_interface/include/controller_interface/chainable_controller_interface.hpp
@@ -88,6 +88,7 @@ class ChainableControllerInterface : public ControllerInterfaceBase
    * \default returns true so the method don't have to be overridden if controller can always switch
    * chained mode.
    */
+  CONTROLLER_INTERFACE_PUBLIC
   virtual bool on_set_chained_mode(bool chained_mode);
 
   /// Update reference from input topics when not in chained mode.
