diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0e33091..4d5405f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -15,7 +15,7 @@ if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra)
 endif()
 
-find_package(OpenCV REQUIRED COMPONENTS calib3d core highgui imgproc)
+find_package(OpenCV REQUIRED COMPONENTS calib core highgui imgproc)
 find_package(sensor_msgs REQUIRED)
 
 add_library(${PROJECT_NAME}
@@ -26,7 +26,7 @@ target_include_directories(${PROJECT_NAME} PUBLIC
   "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
   "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>")
 target_link_libraries(${PROJECT_NAME} PUBLIC
-  opencv_calib3d
+  opencv_calib
   opencv_core
   opencv_highgui
   opencv_imgproc
diff --git a/include/image_geometry/pinhole_camera_model.h b/include/image_geometry/pinhole_camera_model.h
index fdbecd7..7bc0f92 100644
--- a/include/image_geometry/pinhole_camera_model.h
+++ b/include/image_geometry/pinhole_camera_model.h
@@ -6,7 +6,7 @@
 #include <sensor_msgs/msg/camera_info.hpp>
 #include <opencv2/core/core.hpp>
 #include <opencv2/imgproc/imgproc.hpp>
-#include <opencv2/calib3d/calib3d.hpp>
+#include <opencv2/calib3d.hpp>
 #include <stdexcept>
 #include <string>
 
