diff -ruN a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -13,12 +13,7 @@
 find_package(ament_cmake_auto REQUIRED)
 ament_auto_find_build_dependencies()
 
-find_package(Eigen3 QUIET)
-if(NOT EIGEN3_FOUND)
-  find_package(Eigen REQUIRED)
-  set(EIGEN3_INCLUDE_DIRS ${EIGEN_INCLUDE_DIRS})
-  include_directories(include ${EIGEN3_INCLUDE_DIRS})
-endif()
+find_package(Eigen3 REQUIRED)
 
 find_package(OpenCV REQUIRED)
 
diff -ruN a/src/crop_foremost.cpp b/src/crop_foremost.cpp
--- a/src/crop_foremost.cpp
+++ b/src/crop_foremost.cpp
@@ -128,7 +128,7 @@
     case CV_8UC1:
     case CV_8SC1:
     case CV_32F:
-      cv::threshold(cv_ptr->image, cv_ptr->image, minVal + distance_, 0, CV_THRESH_TOZERO_INV);
+      cv::threshold(cv_ptr->image, cv_ptr->image, minVal + distance_, 0, cv::THRESH_TOZERO_INV);
       break;
     case CV_16UC1:
     case CV_16SC1:
@@ -136,7 +136,7 @@
     case CV_64F:
       // 8 bit or 32 bit floating array is required to use cv::threshold
       cv_ptr->image.convertTo(cv_ptr->image, CV_32F);
-      cv::threshold(cv_ptr->image, cv_ptr->image, minVal + distance_, 1, CV_THRESH_TOZERO_INV);
+      cv::threshold(cv_ptr->image, cv_ptr->image, minVal + distance_, 1, cv::THRESH_TOZERO_INV);
 
       cv_ptr->image.convertTo(cv_ptr->image, imtype);
       break;
