diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7863ddb..5f361df 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -32,6 +32,8 @@ include(IceoryxPoshDeployment)
 
 if(CMAKE_SYSTEM_NAME MATCHES Linux OR CMAKE_SYSTEM_NAME MATCHES Darwin)
     option(BUILD_SHARED_LIBS "Create shared libraries by default" ON)
+else()
+    set(BUILD_SHARED_LIBS OFF CACHE BOOL "" FORCE)
 endif()
 
 set(PREFIX iceoryx/v${CMAKE_PROJECT_VERSION})
