diff --git a/rosidl_cli/completion/rosidl-argcomplete.bash b/rosidl_cli/completion/rosidl-argcomplete.bash
index 0262ec2..c3e8606 100644
--- a/rosidl_cli/completion/rosidl-argcomplete.bash
+++ b/rosidl_cli/completion/rosidl-argcomplete.bash
@@ -12,8 +12,6 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-if type register-python-argcomplete3 > /dev/null 2>&1; then
-  eval "$(register-python-argcomplete3 rosidl)"
-elif type register-python-argcomplete > /dev/null 2>&1; then
+if type register-python-argcomplete > /dev/null 2>&1; then
   eval "$(register-python-argcomplete rosidl)"
 fi
diff --git a/rosidl_cli/completion/rosidl-argcomplete.zsh b/rosidl_cli/completion/rosidl-argcomplete.zsh
index 9a86dbd..c4a4253 100644
--- a/rosidl_cli/completion/rosidl-argcomplete.zsh
+++ b/rosidl_cli/completion/rosidl-argcomplete.zsh
@@ -18,6 +18,6 @@ autoload -U +X bashcompinit && bashcompinit
 # Get this scripts directory
 __rosidl_cli_completion_dir=${0:a:h}
 # Just source the bash version, it works in zsh too
-source "$__rosidl_cli_completion_dir/rosidl-argcomplete.bash"
+source "$CONDA_PREFIX/share/bash-completion/completions/rosidl-argcomplete.bash"
 # Cleanup
 unset __rosidl_cli_completion_dir
diff --git a/rosidl_cli/setup.py b/rosidl_cli/setup.py
index dd12abb..4cd1fcf 100644
--- a/rosidl_cli/setup.py
+++ b/rosidl_cli/setup.py
@@ -23,6 +23,14 @@ setup(
             'completion/rosidl-argcomplete.bash',
             'completion/rosidl-argcomplete.zsh'
         ]),
+
+        # [pixi] install autocompletion scripts
+        ('share/bash-completion/completions', [
+            'completion/rosidl-argcomplete.bash',
+        ]),
+        ('share/zsh/site-functions', [
+            'completion/rosidl-argcomplete.zsh'
+        ]),
     ],
     zip_safe=False,
     author='Michel Hidalgo',
