diff --git a/CMakeLists.txt b/CMakeLists.txt
index f33fc523..4ee0d389 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -51,13 +51,14 @@ set(PROJECT_DEPENDENCIES
     yaml-cpp
     image_transport
     cv_bridge
-    geographic_msgs)
+    geographic_msgs
+    GeographicLib)
 
 foreach(DEPENDENCY ${PROJECT_DEPENDENCIES})
   find_package(${DEPENDENCY} REQUIRED)
 endforeach()
 
-find_package(GeographicLib REQUIRED)
+# find_package(GeographicLib REQUIRED)
 
 # Pybind11
 find_package(Python3 REQUIRED COMPONENTS Interpreter Development)
@@ -82,8 +83,15 @@ set(SOURCE_CPP_FILES
 
 # Create the library
 add_library(${PROJECT_NAME} STATIC ${SOURCE_CPP_FILES})
-target_link_libraries(${PROJECT_NAME} yaml-cpp)
-target_link_libraries(${PROJECT_NAME} ${GeographicLib_LIBRARIES})
+if(TARGET GeographicLib::GeographicLib)
+  set(GEOLIBS GeographicLib::GeographicLib)
+elseif(DEFINED GeographicLib_LIBRARIES)
+  set(GEOLIBS ${GeographicLib_LIBRARIES})
+else()
+  message(FATAL_ERROR "GeographicLib not found (no GeographicLib::GeographicLib target and no GeographicLib_LIBRARIES).")
+endif()
+
+target_link_libraries(${PROJECT_NAME} yaml-cpp ${GEOLIBS})
 
 # Mocks library
 set(MOCKS_LIBRARY_NAME mocks)
diff --git a/package.xml b/package.xml
index cad667e8..d8ddb4c3 100644
--- a/package.xml
+++ b/package.xml
@@ -30,7 +30,10 @@
   <depend>image_transport</depend>
   <depend>cv_bridge</depend>
 
-  <depend>geographiclib</depend>
+  <!-- <depend>geographiclib</depend> -->
+  <build_depend>geographiclib</build_depend>
+  <build_export_depend>geographiclib</build_export_depend>
+  <exec_depend>geographiclib</exec_depend>
   <depend>yaml-cpp</depend>
   <depend>eigen</depend>
   <depend>pybind11-dev</depend>
