diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3e3d417fb..91c15c6b2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,8 +12,7 @@ find_package(fmt REQUIRED)
 find_package(moveit_core REQUIRED)
 find_package(moveit_ros_planning REQUIRED)
 find_package(moveit_ros_visualization REQUIRED)
-find_package(Qt5Core REQUIRED)
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt5 COMPONENTS Core Widgets REQUIRED)
 find_package(pluginlib REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rviz_common REQUIRED)
@@ -43,6 +42,10 @@ add_library(
   src/utilities.cpp
   src/xml_syntax_highlighter.cpp
   ${MOC_FILES})
+
+include(GenerateExportHeader)
+generate_export_header(${PROJECT_NAME})
+
 target_include_directories(
   moveit_setup_framework
   PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
@@ -54,8 +57,7 @@ ament_target_dependencies(
   moveit_ros_planning
   moveit_ros_visualization
   pluginlib
-  Qt5Core
-  Qt5Widgets
+  Qt5
   rclcpp
   rviz_common
   rviz_rendering
@@ -67,6 +69,7 @@ install(FILES moveit_setup_framework_plugins.xml
         DESTINATION share/moveit_setup_framework)

 install(DIRECTORY templates DESTINATION share/moveit_setup_framework)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}_export.h DESTINATION include)

 ament_export_targets(moveit_setup_frameworkTargets HAS_LIBRARY_TARGET)
 install(
diff --git a/include/moveit_setup_framework/qt/setup_step_widget.hpp b/include/moveit_setup_framework/qt/setup_step_widget.hpp
index f32e7b5ad..c6f472682 100644
--- a/include/moveit_setup_framework/qt/setup_step_widget.hpp
+++ b/include/moveit_setup_framework/qt/setup_step_widget.hpp
@@ -40,12 +40,14 @@
 #include <moveit_setup_framework/qt/rviz_panel.hpp>
 #include <QWidget>
 
+#include <moveit_setup_framework_export.h>
+
 namespace moveit_setup
 {
 /**
  * @brief The GUI code for one SetupStep
  */
-class SetupStepWidget : public QWidget
+class MOVEIT_SETUP_FRAMEWORK_EXPORT SetupStepWidget : public QWidget
 {
   Q_OBJECT
 public:
diff --git a/include/moveit_setup_framework/templates.hpp b/include/moveit_setup_framework/templates.hpp
index 1555eee16..b02d52024 100644
--- a/include/moveit_setup_framework/templates.hpp
+++ b/include/moveit_setup_framework/templates.hpp
@@ -38,6 +38,8 @@
 #include <moveit_setup_framework/generated_file.hpp>
 #include <string>
 
+#include <moveit_setup_framework_export.h>
+
 namespace moveit_setup
 {
 /**
@@ -69,7 +71,7 @@ public:
 
   bool write() override;
 
-  static std::vector<TemplateVariable> variables;
+  static MOVEIT_SETUP_FRAMEWORK_EXPORT std::vector<TemplateVariable> variables;
 };
 
 }  // namespace moveit_setup
diff --git a/src/urdf_config.cpp b/src/urdf_config.cpp
index 72d9bf4af..3ca176656 100644
--- a/src/urdf_config.cpp
+++ b/src/urdf_config.cpp
@@ -36,6 +36,7 @@
 #include <moveit_setup_framework/utilities.hpp>
 #include <moveit/rdf_loader/rdf_loader.hpp>
 #include <fmt/format.h>
+#include <fmt/ranges.h>
 
 namespace moveit_setup
 {
