diff --git a/CMakeLists.txt b/CMakeLists.txt
index 36e80da02..ca5263f2d 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,8 +11,7 @@ find_package(ament_index_cpp REQUIRED)
 find_package(moveit_core REQUIRED)
 find_package(moveit_ros_planning REQUIRED)
 find_package(moveit_ros_visualization REQUIRED)
-find_package(Qt5Core REQUIRED)
-find_package(Qt5Widgets REQUIRED)
+find_package(Qt5 COMPONENTS Core Widgets REQUIRED)
 find_package(pluginlib REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(rviz_common REQUIRED)
@@ -42,6 +41,10 @@ add_library(${PROJECT_NAME}
     src/xml_syntax_highlighter.cpp
     ${MOC_FILES}
 )
+
+include(GenerateExportHeader)
+generate_export_header(${PROJECT_NAME})
+
 target_include_directories(${PROJECT_NAME} PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include>
@@ -52,8 +55,7 @@ ament_target_dependencies(${PROJECT_NAME}
   moveit_ros_planning
   moveit_ros_visualization
   pluginlib
-  Qt5Core
-  Qt5Widgets
+  Qt5
   rclcpp
   rviz_common
   rviz_rendering
@@ -73,6 +75,7 @@ install(FILES moveit_setup_framework_plugins.xml
 )
 
 install(DIRECTORY templates DESTINATION share/${PROJECT_NAME})
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}_export.h DESTINATION include)
 
 ament_export_targets(export_moveit_setup_framework HAS_LIBRARY_TARGET)
 install(TARGETS ${PROJECT_NAME}
diff --git a/include/moveit_setup_framework/qt/setup_step_widget.hpp b/include/moveit_setup_framework/qt/setup_step_widget.hpp
index f32e7b5ad..c6f472682 100644
--- a/include/moveit_setup_framework/qt/setup_step_widget.hpp
+++ b/include/moveit_setup_framework/qt/setup_step_widget.hpp
@@ -40,12 +40,14 @@
 #include <moveit_setup_framework/qt/rviz_panel.hpp>
 #include <QWidget>
 
+#include <moveit_setup_framework_export.h>
+
 namespace moveit_setup
 {
 /**
  * @brief The GUI code for one SetupStep
  */
-class SetupStepWidget : public QWidget
+class MOVEIT_SETUP_FRAMEWORK_EXPORT SetupStepWidget : public QWidget
 {
   Q_OBJECT
 public:
diff --git a/include/moveit_setup_framework/templates.hpp b/include/moveit_setup_framework/templates.hpp
index fb90a7570..695f7a9cf 100644
--- a/include/moveit_setup_framework/templates.hpp
+++ b/include/moveit_setup_framework/templates.hpp
@@ -38,6 +38,8 @@
 #include <moveit_setup_framework/generated_file.hpp>
 #include <string>
 
+#include <moveit_setup_framework_export.h>
+
 namespace moveit_setup
 {
 /**
@@ -69,7 +71,7 @@ public:
 
   bool write() override;
 
-  static std::vector<TemplateVariable> variables_;
+  static MOVEIT_SETUP_FRAMEWORK_EXPORT std::vector<TemplateVariable> variables_;
 };
 
 }  // namespace moveit_setup
