diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2387c32..0413704 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -8,7 +8,7 @@ endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   if($ENV{ROS_VERSION} EQUAL 2)
-    add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+    add_compile_options(-Wall -Wextra -Wpedantic)
   endif()
 endif()
 
diff --git a/include/usb_cam/formats/av_pixel_format_helper.hpp b/include/usb_cam/formats/av_pixel_format_helper.hpp
index ad1792e..1ea720d 100644
--- a/include/usb_cam/formats/av_pixel_format_helper.hpp
+++ b/include/usb_cam/formats/av_pixel_format_helper.hpp
@@ -744,11 +744,6 @@ const std::unordered_map<std::string, AVPixelFormat> STR_2_AVPIXFMT = {
                                                                       ///< big-endian
 
 
-  {stringify(AV_PIX_FMT_XVMC), AV_PIX_FMT_XVMC},                      ///< XVideo Motion
-                                                                      ///< Acceleration via common
-                                                                      ///< packet passing
-
-
   {stringify(AV_PIX_FMT_YUV440P10LE), AV_PIX_FMT_YUV440P10LE},        ///< planar YUV 4:4:0,20bpp,
                                                                       ///< (1 Cr & Cb sample per
                                                                       ///< 1x2 Y samples),