diff --git a/io_context/CMakeLists.txt b/io_context/CMakeLists.txt
index 54804aa..3c04653 100644
--- a/io_context/CMakeLists.txt
+++ b/io_context/CMakeLists.txt
@@ -25,6 +25,7 @@ endif()
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 ## dependencies
 find_package(ament_cmake_auto REQUIRED)
