diff --git a/tricycle_controller/CMakeLists.txt b/tricycle_controller/CMakeLists.txt
index cdc69f182a..94fba916b1 100644
--- a/tricycle_controller/CMakeLists.txt
+++ b/tricycle_controller/CMakeLists.txt
@@ -47,6 +47,7 @@ target_include_directories(tricycle_controller PUBLIC
 )
 target_link_libraries(tricycle_controller PUBLIC tricycle_controller_parameters)
 ament_target_dependencies(tricycle_controller PUBLIC ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_compile_definitions(tricycle_controller PRIVATE _USE_MATH_DEFINES)
 
 pluginlib_export_plugin_description_file(controller_interface tricycle_controller.xml)
 
