diff --git a/CMakeLists.txt b/CMakeLists.txt
index 403e82b..62de9d6 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -190,8 +190,13 @@ set_target_properties(iceoryx_utils PROPERTIES
 )
 
 if(LINUX)
+    # Find libacl properly
+    find_package(PkgConfig REQUIRED)
+    pkg_check_modules(ACL REQUIRED libacl)
+
     set(ICEORYX_PLATFORM ${CMAKE_CURRENT_SOURCE_DIR}/platform/linux/)
-    target_link_libraries(iceoryx_utils PRIVATE acl atomic ${CODE_COVERAGE_LIBS})
+    target_include_directories(iceoryx_utils PUBLIC ${ACL_INCLUDE_DIRS})
+    target_link_libraries(iceoryx_utils PRIVATE ${ACL_LIBRARIES} atomic ${CODE_COVERAGE_LIBS})
 elseif(QNX)
     set(ICEORYX_PLATFORM ${CMAKE_CURRENT_SOURCE_DIR}/platform/qnx/)
 elseif(APPLE)
