diff --git a/lib/karto_sdk/CMakeLists.txt b/lib/karto_sdk/CMakeLists.txt
index a620bfd..f1e0f60 100644
--- a/lib/karto_sdk/CMakeLists.txt
+++ b/lib/karto_sdk/CMakeLists.txt
@@ -11,7 +11,7 @@ set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/cmake/")
 find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(Eigen3 REQUIRED)
-find_package(Boost REQUIRED system serialization filesystem thread)
+find_package(Boost REQUIRED serialization filesystem thread)
 find_package(TBB REQUIRED NO_CMAKE_PACKAGE_REGISTRY)
 
 set(dependencies
@@ -29,8 +29,9 @@ add_definitions(${EIGEN3_DEFINITIONS})
 
 include_directories(include ${EIGEN3_INCLUDE_DIRS} ${TBB_INCLUDE_DIRS} ${Boost_INCLUDE_DIRS})
 add_library(kartoSlamToolbox SHARED src/Karto.cpp src/Mapper.cpp)
+target_compile_definitions(kartoSlamToolbox PRIVATE KARTO_DYNAMIC)
 ament_target_dependencies(kartoSlamToolbox ${dependencies})
-target_link_libraries(kartoSlamToolbox ${Boost_LIBRARIES} TBB::tbb)
+target_link_libraries(kartoSlamToolbox Eigen3::Eigen ${Boost_LIBRARIES} TBB::tbb)
 
 install(DIRECTORY include/
 	DESTINATION include/
diff --git a/CMake/FindCSparse.cmake b/CMake/FindCSparse.cmake
index 2da6c12..9c4917d 100644
--- a/CMake/FindCSparse.cmake
+++ b/CMake/FindCSparse.cmake
@@ -1,5 +1,6 @@
 # Look for csparse; note the difference in the directory specifications!
 FIND_PATH(CSPARSE_INCLUDE_DIR NAMES cs.h
+  PATH_SUFFIXES suitesparse
   PATHS
   /usr/include/suitesparse
   /usr/include
