diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6d47b08..1b0c4f0 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,6 +37,21 @@ target_link_libraries(rsl PUBLIC
     tl_expected::tl_expected
 )
 
+# There is no explicit export of symbols in the library either via
+# hand-written ***_export.h headers or generate_export_header CMake macro,
+# as the header-only functions in this library are quite limited in number,
+# it is perfectly ok to export all of them (as done in *nix) with the
+# WINDOWS_EXPORT_ALL_SYMBOLS property
+if(MSVC)
+    set_target_properties(rsl PROPERTIES
+        WINDOWS_EXPORT_ALL_SYMBOLS TRUE
+    )
+    # On Windows, also ensure that all .dll libraries are placed in the
+    # same build directory so they can be found by the loader (there is
+    # no rpath on Windows)
+    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/${CMAKE_INSTALL_BINDIR}")
+endif()
+
 add_subdirectory(docs)
 
 option(RSL_BUILD_TESTING "Build tests" OFF)
