diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5d1c744..4912b76 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -23,7 +23,7 @@ else()
         -Wno-deprecated-declarations -Wno-ignored-attributes -Wno-unused-parameter
         -Wnon-virtual-dtor -Wno-unused-variable -Wnull-dereference -Wold-style-cast
         -Woverloaded-virtual -Wshadow -Wsign-conversion
-        -Werror -Wpedantic)
+        -Wpedantic)
 endif()
 
