diff --git a/include/admittance_controller/admittance_controller.hpp b/include/admittance_controller/admittance_controller.hpp
index b6473c6ff..3f1ccadbd 100644
--- a/include/admittance_controller/admittance_controller.hpp
+++ b/include/admittance_controller/admittance_controller.hpp
@@ -95,8 +95,10 @@ public:
     const rclcpp::Time & time, const rclcpp::Duration & period) override;
 
 protected:
+  ADMITTANCE_CONTROLLER_PUBLIC
   std::vector<hardware_interface::CommandInterface> on_export_reference_interfaces() override;
 
+  ADMITTANCE_CONTROLLER_PUBLIC
   controller_interface::return_type update_reference_from_subscribers() override;
 
   size_t num_joints_ = 0;
@@ -168,6 +170,7 @@ protected:
    * @brief Read values from hardware interfaces and set corresponding fields of state_current and
    * ft_values
    */
+  ADMITTANCE_CONTROLLER_PUBLIC
   void read_state_from_hardware(
     trajectory_msgs::msg::JointTrajectoryPoint & state_current,
     geometry_msgs::msg::Wrench & ft_values);
@@ -176,11 +179,13 @@ protected:
    * @brief Set fields of state_reference with values from controllers exported position and
    * velocity references
    */
+  ADMITTANCE_CONTROLLER_PUBLIC
   void read_state_reference_interfaces(trajectory_msgs::msg::JointTrajectoryPoint & state);
 
   /**
    * @brief Write values from state_command to claimed hardware interfaces
    */
+  ADMITTANCE_CONTROLLER_PUBLIC
   void write_state_to_hardware(const trajectory_msgs::msg::JointTrajectoryPoint & state_command);
 };
 
