diff --git a/test_wasm/CMakeLists.txt b/test_wasm/CMakeLists.txt
index c6c95b9..0c6e621 100644
--- a/test_wasm/CMakeLists.txt
+++ b/test_wasm/CMakeLists.txt
@@ -10,21 +10,22 @@ find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(std_msgs REQUIRED)
 find_package(example_interfaces REQUIRED)
+find_package(Python COMPONENTS Interpreter Development REQUIRED)
 
 # Targets and link
 include(cmake/WasmBuildOptions.cmake)
 
 add_executable(talker src/test_publisher.cpp)
-ament_target_dependencies(talker rclcpp std_msgs)
+target_link_libraries(talker PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
 
 add_executable(listener src/test_subscriber.cpp)
-ament_target_dependencies(listener rclcpp std_msgs)
+target_link_libraries(listener PUBLIC ${std_msgs_TARGETS} rclcpp::rclcpp Python::Python)
 
 add_executable(server src/test_service_server.cpp)
-ament_target_dependencies(server rclcpp example_interfaces)
+target_link_libraries(server PUBLIC example_interfaces::example_interfaces__rosidl_typesupport_c example_interfaces::example_interfaces__rosidl_typesupport_cpp rclcpp::rclcpp Python::Python)
 
 add_executable(client src/test_service_client.cpp)
-ament_target_dependencies(client rclcpp example_interfaces)
+target_link_libraries(client PUBLIC example_interfaces::example_interfaces__rosidl_typesupport_c example_interfaces::example_interfaces__rosidl_typesupport_cpp rclcpp::rclcpp Python::Python)
 
 add_compile_options("SHELL:-s USE_PTHREADS")
 
diff --git a/test_wasm/cmake/WasmBuildOptions.cmake b/test_wasm/cmake/WasmBuildOptions.cmake
index e6f4aab..4cfd775 100644
--- a/test_wasm/cmake/WasmBuildOptions.cmake
+++ b/test_wasm/cmake/WasmBuildOptions.cmake
@@ -18,7 +18,7 @@ function(wasm_link_options)
             PUBLIC "SHELL: -fexceptions"
             PUBLIC "SHELL: -s ASYNCIFY=1"
             PUBLIC "SHELL: -s ALLOW_MEMORY_GROWTH=1"
-            PUBLIC "SHELL: -s DEMANGLE_SUPPORT=1"
+            # PUBLIC "SHELL: -s DEMANGLE_SUPPORT=1"
             PUBLIC "SHELL: -s ASSERTIONS=1"
             PUBLIC "SHELL: -s EXIT_RUNTIME=1"
             PUBLIC "SHELL: -s WASM=1"
@@ -26,7 +26,8 @@ function(wasm_link_options)
             PUBLIC "SHELL: -s TOTAL_STACK=16mb"
             PUBLIC "SHELL: -s INITIAL_MEMORY=64mb"
             PUBLIC "SHELL: -s ASYNCIFY_STACK_SIZE=24576"
-            PUBLIC "SHELL: -03" # Optimize for async
+            PUBLIC "SHELL: -03" # Optimize for async
+            PUBLIC "SHELL: -s MINIFY_HTML=0"
         )
     endforeach(arg IN LISTS ARGN)
 endfunction()
