diff --git a/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server.hpp b/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server.hpp
index ec9779444..4909367f9 100644
--- a/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server.hpp
+++ b/nav2_behavior_tree/include/nav2_behavior_tree/bt_action_server.hpp
@@ -168,7 +168,7 @@ class BtActionServer
    * @brief Getter function for the current BT tree
    * @return BT::Tree Current behavior tree
    */
-  BT::Tree getTree() const
+  BT::Tree& getTree()
   {
     return tree_;
   }
