diff --git a/CMakeLists.txt b/CMakeLists.txt
index eb50d67..6be3b86 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -11,9 +11,16 @@ ament_auto_add_library(autoware_test_utils SHARED
   src/mock_data_parser.cpp
 )
 target_link_libraries(autoware_test_utils
-  yaml-cpp
+  yaml-cpp::yaml-cpp
 )
 
+if(NOT MSVC)
+  target_compile_options(autoware_test_utils
+    PUBLIC
+      -Wno-deprecated-declarations
+  )
+endif()
+
 ament_auto_add_executable(topic_snapshot_saver src/topic_snapshot_saver.cpp)
 
 target_link_libraries(topic_snapshot_saver autoware_test_utils yaml-cpp)
