diff --git a/robot_controllers_interface/CMakeLists.txt b/robot_controllers_interface/CMakeLists.txt
index 2c421fa..822b755 100644
--- a/robot_controllers_interface/CMakeLists.txt
+++ b/robot_controllers_interface/CMakeLists.txt
@@ -14,6 +14,7 @@ endif()
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
 endif()
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 find_package(ament_cmake REQUIRED)
 find_package(pluginlib REQUIRED)
