diff --git a/CMakeLists.txt b/CMakeLists.txt
index b1c4e83..9d3a4a2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,22 +26,7 @@ if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release)
 endif()
 
-# TODO(henningkayser): Remove policy fix when assimp 5.1 is available
-# Suppress policy warning in assimp (https://github.com/assimp/assimp/pull/2722)
-set(CMAKE_POLICY_DEFAULT_CMP0012 NEW)
-find_package(ASSIMP QUIET)
-if(NOT ASSIMP_FOUND)
-  find_package(PkgConfig REQUIRED)
-  # assimp is required, so REQUIRE the second attempt
-  pkg_check_modules(ASSIMP_PC REQUIRED assimp)
-  set(ASSIMP_INCLUDE_DIRS ${ASSIMP_PC_INCLUDE_DIRS})
-endif()
-
-# find *absolute* paths to ASSIMP_LIBRARIES
-# Both, pkg-config and assimp's cmake-config don't provide an absolute library path.
-# For, pkg-config the path is in ASSIMP_PC_LIBRARY_DIRS, for cmake in ASSIMP_LIBRARY_DIRS.
-find_library(ASSIMP_ABS_LIBRARIES NAMES ${ASSIMP_LIBRARIES} assimp HINTS ${ASSIMP_LIBRARY_DIRS} ${ASSIMP_PC_LIBRARY_DIRS})
-set(ASSIMP_LIBRARIES "${ASSIMP_ABS_LIBRARIES}")
+find_package(assimp REQUIRED)
 
 # These need to be in this order to find header files
 find_package(eigen3_cmake_module REQUIRED)
@@ -75,6 +60,7 @@ set(THIS_PACKAGE_EXPORT_DEPENDS
   resource_retriever
   shape_msgs
   visualization_msgs
+  assimp
 )
 
 # Set VERSION from package.xml
@@ -97,9 +83,9 @@ target_compile_options(${PROJECT_NAME} PRIVATE ${PROJECT_COMPILE_OPTIONS})
 ament_target_dependencies(${PROJECT_NAME}
   ${THIS_PACKAGE_EXPORT_DEPENDS}
 )
+target_link_libraries(${PROJECT_NAME} assimp::assimp)
 ament_target_dependencies(${PROJECT_NAME} SYSTEM
   # We don't export these dependencies because their cmake is broken
-  ASSIMP
   QHULL
 )
 
