diff --git a/src/interface.cpp b/src/interface.cpp
index 16e2031..26108bc 100644
--- a/src/interface.cpp
+++ b/src/interface.cpp
@@ -21,7 +21,7 @@ InterfaceNode::InterfaceNode(const rclcpp::NodeOptions & options)
 : Node("interface", options),
   srv_(create_service<Version::Service>(
     Version::name,
-    [this](
+    [](
       const Version::Service::Request::SharedPtr, const Version::Service::Response::SharedPtr res) {
       res->major = 1;
       res->minor = 9;
