diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1899f1e..2a5ce73 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -16,7 +16,11 @@
 cmake_minimum_required(VERSION 3.16)
 project(apriltag_detector_umich)
 
-add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+endif()
+set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+add_definitions(-DNOMINMAX)
 
 # find dependencies
 find_package(ament_cmake REQUIRED)
@@ -64,8 +68,7 @@ ament_export_libraries(${PROJECT_NAME})
 
 install(
   TARGETS ${PROJECT_NAME}
-  EXPORT ${PROJECT_NAME}_export
-  DESTINATION lib)
+  EXPORT ${PROJECT_NAME}_export)
 
 if(BUILD_TESTING)
   find_package(ament_cmake REQUIRED)
