diff --git a/cmake/Modules/FindTinyXML2.cmake b/cmake/Modules/FindTinyXML2.cmake
index 87991be..5436d86 100644
--- a/cmake/Modules/FindTinyXML2.cmake
+++ b/cmake/Modules/FindTinyXML2.cmake
@@ -67,6 +67,12 @@ else()
     set_property(TARGET tinyxml2::tinyxml2 PROPERTY INTERFACE_INCLUDE_DIRECTORIES ${TINYXML2_INCLUDE_DIR})
     list(APPEND TinyXML2_TARGETS tinyxml2::tinyxml2)
   endif()
+
+  # Compatibility with https://github.com/gazebosim/gz-cmake/blob/eb1c510e6278935eb742ed92c6a6d1388439f8bd/cmake/FindTINYXML2.cmake#L4
+  if(NOT TARGET TINYXML2::TINYXML2)
+      add_library(TINYXML2::TINYXML2 INTERFACE IMPORTED)
+      set_property(TARGET TINYXML2::TINYXML2 PROPERTY INTERFACE_LINK_LIBRARIES tinyxml2::tinyxml2)
+  endif()
 endif()
 
 # Set mixed case INCLUDE_DIRS and LIBRARY variables from upper case ones.
