diff --git a/include/diff_drive_controller/diff_drive_controller.hpp b/include/diff_drive_controller/diff_drive_controller.hpp
index 3584a1d1f..3c1336fb1 100644
--- a/include/diff_drive_controller/diff_drive_controller.hpp
+++ b/include/diff_drive_controller/diff_drive_controller.hpp
@@ -98,6 +98,7 @@ protected:
     std::reference_wrapper<hardware_interface::LoanedCommandInterface> velocity;
   };
 
+  DIFF_DRIVE_CONTROLLER_PUBLIC
   const char * feedback_type() const;
   controller_interface::CallbackReturn configure_side(
     const std::string & side, const std::vector<std::string> & wheel_names,
@@ -152,7 +153,9 @@ protected:
   bool is_halted = false;
   bool use_stamped_vel_ = true;
 
+  DIFF_DRIVE_CONTROLLER_PUBLIC
   bool reset();
+  DIFF_DRIVE_CONTROLLER_PUBLIC
   void halt();
 };
 }  // namespace diff_drive_controller
