diff --git a/CMakeLists.txt b/CMakeLists.txt
index a3b46942b..e228086fa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,6 +1,8 @@
 cmake_minimum_required(VERSION 3.5)
 project(nav2_waypoint_follower)
 
+add_definitions(-DNOMINMAX)
+
 # Try for OpenCV 4.X, but settle for whatever is installed
 find_package(OpenCV 4 QUIET)
 if(NOT OpenCV_FOUND)

diff --git a/include/nav2_waypoint_follower/plugins/photo_at_waypoint.hpp b/include/nav2_waypoint_follower/plugins/photo_at_waypoint.hpp
index 9c46fdf1f8..1e6048aefc 100644
--- a/include/nav2_waypoint_follower/plugins/photo_at_waypoint.hpp
+++ b/include/nav2_waypoint_follower/plugins/photo_at_waypoint.hpp
@@ -32,8 +32,8 @@
 
 #include "sensor_msgs/msg/image.hpp"
 #include "nav2_core/waypoint_task_executor.hpp"
-#include "opencv4/opencv2/core.hpp"
-#include "opencv4/opencv2/opencv.hpp"
+#include "opencv2/core.hpp"
+#include "opencv2/opencv.hpp"
 #include "cv_bridge/cv_bridge.h"
 #include "image_transport/image_transport.hpp"
 
