diff --git a/include/autoware/ground_filter/grid.hpp b/include/autoware/ground_filter/grid.hpp
index b69426c..c57fe92 100644
--- a/include/autoware/ground_filter/grid.hpp
+++ b/include/autoware/ground_filter/grid.hpp
@@ -25,6 +25,12 @@
 #include <utility>
 #include <vector>
 
+#ifndef M_PIf
+#define M_PIf  3.14159265358979324f
+#define M_PI_2f 2 * M_PIf
+#define M_PI_4f 4 * M_PIf
+#endif
+
 namespace
 {
 
