diff --git a/kinematics_plugin_loader/CMakeLists.txt b/kinematics_plugin_loader/CMakeLists.txt
index ee1b58e8..56858550 100644
--- a/kinematics_plugin_loader/CMakeLists.txt
+++ b/kinematics_plugin_loader/CMakeLists.txt
@@ -1,5 +1,8 @@
 set(MOVEIT_LIB_NAME moveit_kinematics_plugin_loader)
 
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(LIBFCL REQUIRED "fcl>=0.5.0")
+
 add_library(${MOVEIT_LIB_NAME} SHARED src/kinematics_plugin_loader.cpp)
 set_target_properties(${MOVEIT_LIB_NAME} PROPERTIES VERSION "${${PROJECT_NAME}_VERSION}")
 ament_target_dependencies(${MOVEIT_LIB_NAME}
@@ -12,6 +15,7 @@ ament_target_dependencies(${MOVEIT_LIB_NAME}
 )
 target_link_libraries(${MOVEIT_LIB_NAME}
   moveit_rdf_loader
+  ${LIBFCL_LINK_LIBRARIES}
 )
 
 install(DIRECTORY include/ DESTINATION include)
diff --git a/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp b/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp
index f718f3f4..9db5cbb8 100644
--- a/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp
+++ b/kinematics_plugin_loader/src/kinematics_plugin_loader.cpp
@@ -38,6 +38,8 @@
 #include <moveit/rdf_loader/rdf_loader.h>
 #include <pluginlib/class_loader.hpp>
 #include <boost/thread/mutex.hpp>
+#include <boost/bind/placeholders.hpp>
+#include <boost/bind/bind.hpp>
 #include <sstream>
 #include <vector>
 #include <map>
