diff --git a/CMakeLists.txt b/CMakeLists.txt
index bc98189..71ce465 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,9 @@ find_package(geometry_msgs REQUIRED)
 find_package(polygon_msgs REQUIRED)
 
 add_library(polygon_utils src/polygon_utils.cpp)
+if(WIN32 AND MSVC)
+  set_target_properties(polygon_utils PROPERTIES WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
 target_include_directories(polygon_utils PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include>
