diff --git a/rosidl_typesupport_fastrtps_c/rosidl_typesupport_fastrtps_c-extras.cmake.in b/rosidl_typesupport_fastrtps_c/rosidl_typesupport_fastrtps_c-extras.cmake.in
index f23e9d0..cc19345 100644
--- a/rosidl_typesupport_fastrtps_c/rosidl_typesupport_fastrtps_c-extras.cmake.in
+++ b/rosidl_typesupport_fastrtps_c/rosidl_typesupport_fastrtps_c-extras.cmake.in
@@ -21,7 +21,7 @@ else()
     "${rosidl_typesupport_fastrtps_c_BIN}")
 
   set(rosidl_typesupport_fastrtps_c_GENERATOR_FILES
-    "${rosidl_typesupport_fastrtps_c_DIR}/../../../@PYTHON_INSTALL_DIR@/rosidl_typesupport_fastrtps_c/__init__.py")
+    "@PYTHON_INSTALL_DIR@/rosidl_typesupport_fastrtps_c/__init__.py")
   normalize_path(rosidl_typesupport_fastrtps_c_GENERATOR_FILES
     "${rosidl_typesupport_fastrtps_c_GENERATOR_FILES}")
 
