diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0452e0485..38b72376a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -17,8 +17,10 @@ if(WITH_OPENGL)
 
   set(GL_LIBS ${GL_LIBS} ${OPENGL_LIBRARIES})
   if(APPLE)
-    find_package(FreeGLUT REQUIRED)
-    set(SYSTEM_GL_LIBRARIES ${GLEW_LIBRARIES} GLEW::GLEW FreeGLUT::freeglut)
+    list(INSERT CMAKE_FRAMEWORK_PATH 0 /System/Library/Frameworks)
+    find_package(GLUT REQUIRED)
+    # find_package(FreeGLUT REQUIRED)
+    set(SYSTEM_GL_LIBRARIES ${GLEW_LIBRARIES} GLEW::GLEW GLUT::GLUT)
   else()
     find_package(GLUT REQUIRED)
     if(WIN32)
diff --git a/mesh_filter/src/gl_renderer.cpp b/mesh_filter/src/gl_renderer.cpp
index e528cb5b5..9d1689cf6 100644
--- a/mesh_filter/src/gl_renderer.cpp
+++ b/mesh_filter/src/gl_renderer.cpp
@@ -37,11 +37,13 @@
 #include <GL/glew.h>
 #ifdef __APPLE__
 #include <OpenGL/glu.h>
+#include <OpenGL/gl.h>
+#include <GLUT/glut.h>
 #else
 #include <GL/glu.h>
 #include <GL/glut.h>
-#endif
 #include <GL/freeglut.h>
+#endif
 #include <moveit/mesh_filter/gl_renderer.hpp>
 #include <moveit/utils/logger.hpp>
 #include <sstream>
@@ -405,8 +407,13 @@ void mesh_filter::GLRenderer::createGLContext()
     glutIconifyWindow();
     glutHideWindow();
 
-    for (int i = 0; i < 10; ++i)
-      glutMainLoopEvent();
+    for (int i = 0; i < 10; ++i){
+      #ifdef __APPLE__
+        glutCheckLoop();
+      #else
+        glutMainLoopEvent();
+      #endif
+    }
 
     s_context.at(thread_id) = std::pair<unsigned, GLuint>(1, window_id);
   }
diff --git a/semantic_world/CMakeLists.txt b/semantic_world/CMakeLists.txt
index dc8d941da..0baff31eb 100644
--- a/semantic_world/CMakeLists.txt
+++ b/semantic_world/CMakeLists.txt
@@ -13,6 +13,8 @@ ament_target_dependencies(
   moveit_msgs
   tf2_eigen
   Eigen3
-  Boost)
+  Boost
+  OpenCV)
+  
 
 install(DIRECTORY include/ DESTINATION include/moveit_ros_perception)
