From d236b8e9172705076e1c0aba34940307a925d3e2 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio@traversaro.it>
Date: Wed, 18 Dec 2024 13:50:38 +0100
Subject: [PATCH] Drop apt-specific workaround on the octomap version

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8c46189..277356c 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,7 +45,7 @@ find_package(eigen_stl_containers REQUIRED)
 find_package(geometry_msgs REQUIRED)
 # Enforce the system package version on Ubuntu jammy and noble which is also used by libfcl-dev
 # The version is fixed to prevent ABI conflicts with ros-octomap
-find_package(octomap 1.9.7...<1.10.0 REQUIRED)
+find_package(octomap REQUIRED)
 find_package(QHULL REQUIRED)
 find_package(random_numbers REQUIRED)
 find_package(rclcpp REQUIRED)
