diff --git a/CMakeLists.txt b/CMakeLists.txt
index 57ec3424..44ca8e5a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -85,11 +85,6 @@ set(THIS_PACKAGE_INCLUDE_DEPENDS
 ## Libraries ##
 ###############
 
-add_library(planning_context_loader_base SHARED
-  src/planning_context_loader.cpp
-)
-ament_target_dependencies(planning_context_loader_base ${THIS_PACKAGE_INCLUDE_DEPENDS})
-
 add_library(joint_limits_common SHARED
   src/joint_limits_aggregator.cpp
   src/joint_limits_container.cpp
@@ -100,12 +95,19 @@ add_library(joint_limits_common SHARED
 )
 ament_target_dependencies(joint_limits_common ${THIS_PACKAGE_INCLUDE_DEPENDS})
 
+add_library(planning_context_loader_base SHARED
+  src/planning_context_loader.cpp
+)
+ament_target_dependencies(planning_context_loader_base ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(planning_context_loader_base joint_limits_common)
+
 add_library(trajectory_generation_common SHARED
   src/trajectory_functions.cpp
   src/trajectory_generator.cpp
   src/trajectory_blender_transition_window.cpp
 )
 ament_target_dependencies(trajectory_generation_common ${THIS_PACKAGE_INCLUDE_DEPENDS})
+target_link_libraries(trajectory_generation_common joint_limits_common)
 
 add_library(command_list_manager SHARED
   src/command_list_manager.cpp
