diff --git a/forward_command_controller/include/forward_command_controller/forward_command_controller.hpp b/forward_command_controller/include/forward_command_controller/forward_command_controller.hpp
index 91e3aae480..5c2de9131b 100644
--- a/forward_command_controller/include/forward_command_controller/forward_command_controller.hpp
+++ b/forward_command_controller/include/forward_command_controller/forward_command_controller.hpp
@@ -44,7 +44,9 @@ class ForwardCommandController : public ForwardControllersBase
   ForwardCommandController();
 
 protected:
+  FORWARD_COMMAND_CONTROLLER_PUBLIC
   void declare_parameters() override;
+  FORWARD_COMMAND_CONTROLLER_PUBLIC
   controller_interface::CallbackReturn read_parameters() override;
 
   std::shared_ptr<ParamListener> param_listener_;
diff --git a/forward_command_controller/include/forward_command_controller/multi_interface_forward_command_controller.hpp b/forward_command_controller/include/forward_command_controller/multi_interface_forward_command_controller.hpp
index fd7c0d480e..3881c1c9d4 100644
--- a/forward_command_controller/include/forward_command_controller/multi_interface_forward_command_controller.hpp
+++ b/forward_command_controller/include/forward_command_controller/multi_interface_forward_command_controller.hpp
@@ -44,7 +44,9 @@ class MultiInterfaceForwardCommandController
   MultiInterfaceForwardCommandController();
 
 protected:
+  FORWARD_COMMAND_CONTROLLER_PUBLIC
   void declare_parameters() override;
+  FORWARD_COMMAND_CONTROLLER_PUBLIC
   controller_interface::CallbackReturn read_parameters() override;
 
   using Params = multi_interface_forward_command_controller::Params;
