diff --git a/CMakeLists.txt b/CMakeLists.txt
index e725903..a0asd0sa 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -73,6 +73,5 @@ set(libraries mppi_controller mppi_critics)

 foreach(lib IN LISTS libraries)
-  target_compile_options(${lib} PUBLIC -fconcepts)
   target_include_directories(${lib} PUBLIC include ${xsimd_INCLUDE_DIRS} ${OpenMP_INCLUDE_DIRS})
   target_link_libraries(${lib} xtensor xtensor::optimize xtensor::use_xsimd)
 
   option(COVERAGE_ENABLED "Enable code coverage" FALSE)
