diff --git a/moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/action_based_controller_handle.h b/moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/action_based_controller_handle.h
index 311d51f7a5..e6e21efe97 100644
--- a/moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/action_based_controller_handle.h
+++ b/moveit_plugins/moveit_simple_controller_manager/include/moveit_simple_controller_manager/action_based_controller_handle.h
@@ -145,7 +145,7 @@ class ActionBasedControllerHandle : public ActionBasedControllerHandleBase
         do
         {
           status = result_future.wait_for(50ms);
-          if ((status == std::future_status::timeout) and ((node_->now() - start) > timeout))
+          if ((status == std::future_status::timeout) && ((node_->now() - start) > timeout))
           {
             RCLCPP_WARN(LOGGER, "waitForExecution timed out");
             return false;
