diff --git a/cmake/grid_map_package.cmake b/cmake/grid_map_package.cmake
index 4867a92f4..291c9f3ac 100644
--- a/cmake/grid_map_package.cmake
+++ b/cmake/grid_map_package.cmake
@@ -17,7 +17,7 @@ macro(grid_map_package)
   endif()
 
   if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    add_compile_options(-Wall -Wextra -Wpedantic -Werror -Wdeprecated -fPIC)
+    add_compile_options(-Wall -Wextra -Wpedantic -Wdeprecated -fPIC)
   endif()
 
   if(COVERAGE_ENABLED)
