diff --git a/CMakeLists.txt b/CMakeLists.txt
index e4828bf4..1ea63296 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,6 +3,10 @@ set(MRT_PKG_VERSION 4.0.0)
 cmake_minimum_required(VERSION 3.5.1)
 project(lanelet2_io)
 
+if(WIN32 AND MSVC)
+  set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
+endif()
+
 ###################
 ## Find packages ##
 ###################
@@ -52,6 +56,10 @@ mrt_add_library(${PROJECT_NAME}
     SOURCES ${PROJECT_SOURCE_FILES_SRC}
     )
 
+target_compile_definitions(${PROJECT_NAME}
+    PUBLIC "$<$<CXX_COMPILER_ID:MSVC>:_USE_MATH_DEFINES>"
+    )
+
 #############
 ## Install ##
 #############
