diff --git a/CMakeLists.txt b/CMakeLists.txt
index 96b399a4..7e5d3a27 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -225,12 +225,12 @@ add_library(rviz_default_plugins SHARED
 target_include_directories(rviz_default_plugins PUBLIC
   $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
   $<INSTALL_INTERFACE:include/${PROJECT_NAME}>
-  ${Qt5Widgets_INCLUDE_DIRS}
 )
 
 target_link_libraries(rviz_default_plugins PUBLIC
   rviz_ogre_vendor::OgreMain
   rviz_ogre_vendor::OgreOverlay
+  Qt5::Widgets
 )
 
 target_link_libraries(rviz_default_plugins PRIVATE
