diff --git a/test_wasm/CMakeLists.txt b/test_wasm/CMakeLists.txt
index c6c95b9..0c6e621 100644
--- a/test_wasm/CMakeLists.txt
+++ b/test_wasm/CMakeLists.txt
@@ -10,21 +10,43 @@ find_package(ament_cmake REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(std_msgs REQUIRED)
 find_package(example_interfaces REQUIRED)
+find_package(PkgConfig REQUIRED)
+find_package(Python COMPONENTS Interpreter Development REQUIRED)
+find_package(rmw_wasm_cpp REQUIRED)
+find_package(BZip2 REQUIRED)
+find_package(SQLite3 REQUIRED)
+find_package(ZLIB REQUIRED)
+pkg_check_modules(yaml REQUIRED yaml-0.1)
+pkg_check_modules(libffi REQUIRED libffi)
+
+# Hack: avoid double link of libyaml-cpp
+foreach(item ${rmw_wasm_cpp_LIBRARIES})
+    if(item MATCHES "libyaml-cpp.a$")
+        list(REMOVE_ITEM rmw_wasm_cpp_LIBRARIES ${item})
+    endif()
+endforeach()
 
 # Targets and link
 include(cmake/WasmBuildOptions.cmake)
 
+link_directories(${yaml_LIBRARY_DIRS})
+
 add_executable(talker src/test_publisher.cpp)
-ament_target_dependencies(talker rclcpp std_msgs)
+target_link_libraries(talker PUBLIC ${std_msgs_TARGETS} ${rmw_wasm_cpp_LIBRARIES} rclcpp::rclcpp ${libffi_LINK_LIBRARIES} Python::Python ${SQLite3_LIBRARIES} BZip2::BZip2 ZLIB::ZLIB)
 
 add_executable(listener src/test_subscriber.cpp)
-ament_target_dependencies(listener rclcpp std_msgs)
+target_link_libraries(listener PUBLIC ${std_msgs_TARGETS} ${rmw_wasm_cpp_LIBRARIES} rclcpp::rclcpp ${libffi_LINK_LIBRARIES} Python::Python ${SQLite3_LIBRARIES} BZip2::BZip2 ZLIB::ZLIB)
 
 add_executable(server src/test_service_server.cpp)
-ament_target_dependencies(server rclcpp example_interfaces)
+target_link_libraries(server PUBLIC ${rmw_wasm_cpp_LIBRARIES} example_interfaces::example_interfaces__rosidl_typesupport_c example_interfaces::example_interfaces__rosidl_typesupport_cpp rclcpp::rclcpp ${libffi_LINK_LIBRARIES} Python::Python ${SQLite3_LIBRARIES} BZip2::BZip2 ZLIB::ZLIB)
 
 add_executable(client src/test_service_client.cpp)
-ament_target_dependencies(client rclcpp example_interfaces)
+target_link_libraries(client PUBLIC ${rmw_wasm_cpp_LIBRARIES} example_interfaces::example_interfaces__rosidl_typesupport_c example_interfaces::example_interfaces__rosidl_typesupport_cpp rclcpp::rclcpp ${libffi_LINK_LIBRARIES} Python::Python ${SQLite3_LIBRARIES} BZip2::BZip2 ZLIB::ZLIB)
+
+target_compile_options(talker PUBLIC -fPIC)
+target_compile_options(listener PUBLIC -fPIC)
+target_compile_options(server PUBLIC -fPIC)
+target_compile_options(client PUBLIC -fPIC)
 
 add_compile_options("SHELL:-s USE_PTHREADS")
 
